diff -aur busybox-1.19.2.orig//include/libbb.h busybox-1.19.2//include/libbb.h
--- busybox-1.19.2.orig//include/libbb.h	2011-09-16 01:12:38.000000000 +0000
+++ busybox-1.19.2//include/libbb.h	2011-09-16 01:13:21.000000000 +0000
@@ -37,6 +37,7 @@
 #include <sys/mman.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
+#include <sys/sysinfo.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #ifndef major
Only in busybox-1.19.2//include: libbb.h.orig
diff -aur busybox-1.19.2.orig//include/platform.h busybox-1.19.2//include/platform.h
--- busybox-1.19.2.orig//include/platform.h	2011-09-16 01:12:38.000000000 +0000
+++ busybox-1.19.2//include/platform.h	2011-09-16 01:13:21.000000000 +0000
@@ -443,6 +443,13 @@
 # undef HAVE_NET_ETHERNET_H
 #endif
 
+#if defined(__musl__)
+# undef HAVE_SETBIT
+# include <stddef.h>
+# include <termios.h>
+# include <sys/ioctl.h>
+#endif
+
 /*
  * Now, define prototypes for all the functions defined in platform.c
  * These must come after all the HAVE_* macros are defined (or not)
Only in busybox-1.19.2//include: platform.h.orig
diff -aur busybox-1.19.2.orig//miscutils/man.c busybox-1.19.2//miscutils/man.c
--- busybox-1.19.2.orig//miscutils/man.c	2011-09-16 01:12:38.000000000 +0000
+++ busybox-1.19.2//miscutils/man.c	2011-09-16 01:13:21.000000000 +0000
@@ -116,7 +116,7 @@
 	/* "2>&1" is added so that nroff errors are shown in pager too.
 	 * Otherwise it may show just empty screen */
 	cmd = xasprintf(
-		man ? "gtbl | nroff -Tlatin1 -mandoc 2>&1 | %s"
+		man ? "nroff -Tutf -man 2>&1 | %s"
 		    : "%s",
 		pager);
 	system(cmd);
Only in busybox-1.19.2//miscutils: man.c.orig
diff -aur busybox-1.19.2.orig//networking/ifconfig.c busybox-1.19.2//networking/ifconfig.c
--- busybox-1.19.2.orig//networking/ifconfig.c	2011-09-16 01:12:38.000000000 +0000
+++ busybox-1.19.2//networking/ifconfig.c	2011-09-16 01:13:21.000000000 +0000
@@ -56,7 +56,7 @@
 #endif
 
 #if ENABLE_FEATURE_IFCONFIG_SLIP
-# include <net/if_slip.h>
+# include <linux/if_slip.h>
 #endif
 
 /* I don't know if this is needed for busybox or not.  Anyone? */
Only in busybox-1.19.2//networking: ifconfig.c.orig
diff -aur busybox-1.19.2.orig//networking/libiproute/iplink.c busybox-1.19.2//networking/libiproute/iplink.c
--- busybox-1.19.2.orig//networking/libiproute/iplink.c	2011-09-16 01:12:38.000000000 +0000
+++ busybox-1.19.2//networking/libiproute/iplink.c	2011-09-16 01:13:21.000000000 +0000
@@ -5,7 +5,6 @@
  * Licensed under GPLv2 or later, see file LICENSE in this source tree.
  */
 #include <net/if.h>
-#include <net/if_packet.h>
 #include <netpacket/packet.h>
 #include <netinet/if_ether.h>
 
Only in busybox-1.19.2//networking/libiproute: iplink.c.orig
diff -aur busybox-1.19.2.orig//networking/tcpudp.c busybox-1.19.2//networking/tcpudp.c
--- busybox-1.19.2.orig//networking/tcpudp.c	2011-09-16 01:12:38.000000000 +0000
+++ busybox-1.19.2//networking/tcpudp.c	2011-09-16 01:13:21.000000000 +0000
@@ -71,7 +71,7 @@
 /* Wants <limits.h> etc, thus included after libbb.h: */
 #ifdef __linux__
 #include <linux/types.h> /* for __be32 etc */
-#include <linux/netfilter_ipv4.h>
+//#include <linux/netfilter_ipv4.h>
 #endif
 
 // TODO: move into this file:
diff -aur busybox-1.19.2.orig//util-linux/fdisk.c busybox-1.19.2//util-linux/fdisk.c
--- busybox-1.19.2.orig//util-linux/fdisk.c	2011-09-16 01:12:38.000000000 +0000
+++ busybox-1.19.2//util-linux/fdisk.c	2011-09-16 01:13:21.000000000 +0000
@@ -550,7 +550,7 @@
 {
 #if ENABLE_FDISK_SUPPORT_LARGE_DISKS
 	off64_t off = (off64_t)secno * sector_size;
-	if (lseek64(dev_fd, off, SEEK_SET) == (off64_t) -1)
+	if (lseek(dev_fd, off, SEEK_SET) == (off64_t) -1)
 		fdisk_fatal(unable_to_seek);
 #else
 	uint64_t off = (uint64_t)secno * sector_size;
Only in busybox-1.19.2//util-linux: fdisk.c.orig
Make -p count "path components" not slashes, so broken /blah//thing paths work.

diff --git a/editors/patch.c b/editors/patch.c
index 1f2a49b..b1f51cf 100644
--- a/editors/patch.c
+++ b/editors/patch.c
@@ -482,11 +482,11 @@ int patch_main(int argc UNUSED_PARAM, char **argv)
 
 				// handle -p path truncation.
 				for (i=0, s = name; *s;) {
+					char *temp = name;
+
 					if ((option_mask32 & FLAG_PATHLEN) && TT.prefix == i) break;
-					if (*(s++)=='/') {
-						name = s;
-						i++;
-					}
+					while (*(s++)=='/') name = s;
+					if (temp != name) i++;
 				}
 
 				if (empty) {
