diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -91,6 +91,7 @@
 SUBDIR+=	wm
 SUBDIR+=	www
 .endif
+_WRAP_EXTRA_ARGS.CC+=	-DNOFILES_MAX=256
 
 SUBDIR+=	${USER_ADDITIONAL_PKGS}
 
diff --git a/archivers/pax/files/getid.c b/archivers/pax/files/getid.c
--- a/archivers/pax/files/getid.c
+++ b/archivers/pax/files/getid.c
@@ -152,12 +152,14 @@
 	if (!grstart() || !pwstart())
 		return (0);
 
+#if 0
 				/* switch pwcache(3) lookup functions */
 	if (pwcache_groupdb(gi_setgroupent, gi_endgrent,
 			    gi_getgrnam, gi_getgrgid) == -1
 	    || pwcache_userdb(gi_setpassent, gi_endpwent,
 			    gi_getpwnam, gi_getpwuid) == -1)
 		return (0);
+#endif
 
 	return (1);
 }
diff --git a/archivers/pax/files/options.c b/archivers/pax/files/options.c
--- a/archivers/pax/files/options.c
+++ b/archivers/pax/files/options.c
@@ -107,6 +107,8 @@
 #include "mtree.h"
 #endif	/* SMALL */
 
+#include <strings.h>
+
 /*
  * Routines which handle command line options
  */
diff --git a/audio/alsa-lib/distinfo b/audio/alsa-lib/distinfo
--- a/audio/alsa-lib/distinfo
+++ b/audio/alsa-lib/distinfo
@@ -42,3 +42,4 @@
 SHA1 (patch-bk) = 7740eea6a0d4a8ce3db3b93b2af5faacdb9424b1
 SHA1 (patch-include_pcm.h) = 1c458cd7d078356f4a970b0b3592aad139ac0918
 SHA1 (patch-src_control_control__hw.c) = 07cfe3abe586b32f6b989403cd162599da07e865
+SHA1 (patch-qq) = a41ef5695898cde72f31489de2ed10088b0f476f
diff --git a/audio/alsa-lib/patches/patch-qq b/audio/alsa-lib/patches/patch-qq
new file mode 100644
--- /dev/null
+++ b/audio/alsa-lib/patches/patch-qq
@@ -0,0 +1,72 @@
+diff -r bb67315ed01e include/global.h
+--- include/global.h	Mon May 07 21:37:17 2012 +0000
++++ include/global.h	Mon May 07 21:45:14 2012 +0000
+@@ -137,7 +137,7 @@
+ 
+ int snd_user_file(const char *file, char **result);
+ 
+-#if !defined(_POSIX_C_SOURCE) && !defined(_POSIX_SOURCE) && !defined(__NetBSD__) && !defined(__DragonFly__) && !defined(__sun)
++#if !defined(_POSIX_C_SOURCE) && !defined(_POSIX_SOURCE) && !defined(__NetBSD__) && !defined(__DragonFly__) && !defined(__sun) && !defined(_GNU_SOURCE)
+ struct timeval {
+ 	time_t		tv_sec;		/* seconds */
+ 	long		tv_usec;	/* microseconds */
+diff -r bb67315ed01e include/pcm.h
+--- include/pcm.h	Mon May 07 21:37:17 2012 +0000
++++ include/pcm.h	Mon May 07 21:45:14 2012 +0000
+@@ -928,10 +928,10 @@
+ uint32_t snd_pcm_format_silence_32(snd_pcm_format_t format);
+ uint64_t snd_pcm_format_silence_64(snd_pcm_format_t format);
+ #else
+-u_int8_t snd_pcm_format_silence(snd_pcm_format_t format);
+-u_int16_t snd_pcm_format_silence_16(snd_pcm_format_t format);
+-u_int32_t snd_pcm_format_silence_32(snd_pcm_format_t format);
+-u_int64_t snd_pcm_format_silence_64(snd_pcm_format_t format);
++uint8_t snd_pcm_format_silence(snd_pcm_format_t format);
++uint16_t snd_pcm_format_silence_16(snd_pcm_format_t format);
++uint32_t snd_pcm_format_silence_32(snd_pcm_format_t format);
++uint64_t snd_pcm_format_silence_64(snd_pcm_format_t format);
+ #endif
+ int snd_pcm_format_set_silence(snd_pcm_format_t format, void *buf, unsigned int samples);
+ 
+diff -r bb67315ed01e include/sound/asound.h
+--- include/sound/asound.h	Mon May 07 21:37:17 2012 +0000
++++ include/sound/asound.h	Mon May 07 21:45:14 2012 +0000
+@@ -358,7 +358,7 @@
+ #define SNDRV_MASK_MAX	256
+ 
+ struct sndrv_mask {
+-	u_int32_t bits[(SNDRV_MASK_MAX+31)/32];
++	uint32_t bits[(SNDRV_MASK_MAX+31)/32];
+ };
+ 
+ struct sndrv_pcm_hw_params {
+diff -r bb67315ed01e src/dlmisc.c
+--- src/dlmisc.c	Mon May 07 21:37:17 2012 +0000
++++ src/dlmisc.c	Mon May 07 21:45:14 2012 +0000
+@@ -54,11 +54,13 @@
+ #ifdef HAVE_LIBDL
+ 	if (name == NULL) {
+ 		static const char * self = NULL;
++#if 0
+ 		if (self == NULL) {
+ 			Dl_info dlinfo;
+ 			if (dladdr(snd_dlopen, &dlinfo) > 0)
+ 				self = dlinfo.dli_fname;
+ 		}
++#endif
+ 		name = self;
+ 	}
+ #endif
+diff -r fbb5dc5542a8 src/compat/hsearch_r.c
+--- src/compat/hsearch_r.c	Mon May 07 21:52:06 2012 +0000
++++ src/compat/hsearch_r.c	Mon May 07 21:52:21 2012 +0000
+@@ -1,3 +1,4 @@
++#if 0
+ /* Copyright (C) 1993, 1995, 1996, 1997 Free Software Foundation, Inc.
+    This file is part of the GNU C Library.
+    Contributed by Ulrich Drepper <drepper@gnu.ai.mit.edu>, 1993.
+@@ -234,3 +235,4 @@
+   *retval = NULL;
+   return 0;
+ }
++#endif
diff --git a/chat/irssi/Makefile b/chat/irssi/Makefile
--- a/chat/irssi/Makefile
+++ b/chat/irssi/Makefile
@@ -20,9 +20,9 @@
 USE_LIBTOOL=		yes
 USE_TOOLS+=		pkg-config
 
+# [Gregor] Removed --without-ncurses, since there's no other libtermcap
 CONFIGURE_ARGS+=	--with-terminfo \
 			--with-proxy \
-			--without-ncurses \
 			--sysconfdir=${PKG_SYSCONFDIR}
 
 EGDIR=		${PREFIX}/share/examples/irssi
diff --git a/databases/db4/distinfo b/databases/db4/distinfo
--- a/databases/db4/distinfo
+++ b/databases/db4/distinfo
@@ -7,3 +7,4 @@
 SHA1 (patch-ab) = 8f9ecc47527d338e28ee1ce71e694eaf2bdceb92
 SHA1 (patch-ac) = 045a3b83b9c6c164382ec546b06ae248620552cf
 SHA1 (patch-ae) = 43e574154de93dba60392b09d6ebddbca1036849
+SHA1 (patch-aq) = 3baa90e922dc9cb075deee2a2a37d607affead9b
diff --git a/databases/db4/patches/patch-aq b/databases/db4/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/databases/db4/patches/patch-aq
@@ -0,0 +1,11 @@
+diff -r 7a6cd6105754 dist/configure
+--- dist/configure	Sat Apr 28 15:23:22 2012 +0000
++++ dist/configure	Sat Apr 28 15:24:05 2012 +0000
+@@ -18792,6 +18792,7 @@
+ /* end confdefs.h.  */
+ 
+ #include <pthread.h>
++#include <stdlib.h>
+ main() {
+ 	pthread_cond_t cond;
+ 	pthread_mutex_t mutex;
diff --git a/devel/bison/Makefile b/devel/bison/Makefile
--- a/devel/bison/Makefile
+++ b/devel/bison/Makefile
@@ -28,6 +28,7 @@
 # Avoid rebuilding manpage
 pre-build:
 	${TOUCH} ${WRKSRC}/doc/bison.1
+	sed -i 's/#elif 0/#elif 1/' -i ${WRKSRC}/lib/fseterr.c
 
 # "bison" wants a recent version of "gettext" which at least some
 # NetBSD versions don't provide. Figure out whether it will install
diff --git a/devel/bmake/files/suff.c b/devel/bmake/files/suff.c
--- a/devel/bmake/files/suff.c
+++ b/devel/bmake/files/suff.c
@@ -67,6 +67,8 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+ 
+#include <strings.h> /* for ffs */
 
 #ifndef MAKE_NATIVE
 static char rcsid[] = "$NetBSD: suff.c,v 1.4 2009/09/18 21:27:25 joerg Exp $";
diff --git a/devel/bmake/files/targ.c b/devel/bmake/files/targ.c
--- a/devel/bmake/files/targ.c
+++ b/devel/bmake/files/targ.c
@@ -67,6 +67,8 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+ 
+#include <strings.h> /* for ffs */
 
 #ifndef MAKE_NATIVE
 static char rcsid[] = "$NetBSD: targ.c,v 1.5 2011/06/18 22:39:46 bsiegert Exp $";
diff --git a/devel/bmake/files/unit-tests/test.exp b/devel/bmake/files/unit-tests/test.exp
--- a/devel/bmake/files/unit-tests/test.exp
+++ b/devel/bmake/files/unit-tests/test.exp
@@ -27,6 +27,7 @@
 make: "error" line 3: just FYI
 make: "error" line 4: warning: this could be serious
 make: "error" line 5: this is fatal
+*** Error code 1 (ignored)
 UT_DOLLAR=This is $UT_FU
 UT_FOO=foobar is fubar
 UT_FU=fubar
@@ -46,9 +47,9 @@
 At first, I am
 happy
 and now: sad
-.ERROR: Looks like 'sad' is upset.
 *** Error code 1
 
+.ERROR: Looks like 'sad' is upset.
 Stop.
 make: stopped in unit-tests
 simple.1
@@ -75,11 +76,11 @@
 shared.2.1
 shared.2.99
 shared.2.99
+cycle.1.99
 make: Graph cycles through `cycle.2.99'
 make: Graph cycles through `cycle.2.98'
 make: Graph cycles through `cycle.2.97'
 cycle.1.99
-cycle.1.99
 .for with :S;... OK
 b2af338b
 3360ac65
@@ -343,4 +344,3 @@
 six v=is y k=is y
 show-v v=override k=override
 *** Error code 1 (ignored)
-*** Error code 1 (ignored)
diff --git a/devel/flex/PLIST b/devel/flex/PLIST
--- a/devel/flex/PLIST
+++ b/devel/flex/PLIST
@@ -6,19 +6,3 @@
 lib/libfl.a
 lib/libfl_pic.a
 man/man1/flex.1
-share/locale/ca/LC_MESSAGES/flex.mo
-share/locale/da/LC_MESSAGES/flex.mo
-share/locale/de/LC_MESSAGES/flex.mo
-share/locale/es/LC_MESSAGES/flex.mo
-share/locale/fr/LC_MESSAGES/flex.mo
-share/locale/ga/LC_MESSAGES/flex.mo
-share/locale/ko/LC_MESSAGES/flex.mo
-share/locale/nl/LC_MESSAGES/flex.mo
-share/locale/pl/LC_MESSAGES/flex.mo
-share/locale/pt_BR/LC_MESSAGES/flex.mo
-share/locale/ro/LC_MESSAGES/flex.mo
-share/locale/ru/LC_MESSAGES/flex.mo
-share/locale/sv/LC_MESSAGES/flex.mo
-share/locale/tr/LC_MESSAGES/flex.mo
-share/locale/vi/LC_MESSAGES/flex.mo
-share/locale/zh_CN/LC_MESSAGES/flex.mo
diff --git a/devel/glib2/Makefile b/devel/glib2/Makefile
--- a/devel/glib2/Makefile
+++ b/devel/glib2/Makefile
@@ -46,6 +46,12 @@
 
 LIBS.SunOS+=	-lnsl -lsocket
 
+
+
+# [Gregor] bugs in how it uses libdl
+pre-build:
+	sed -i 's/if (msg)/if (!p \&\& msg)/' ${WRKSRC}/gmodule/gmodule-dl.c
+
 .include "../../converters/libiconv/buildlink3.mk"
 .include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../devel/pcre/buildlink3.mk"
diff --git a/devel/gmake/Makefile b/devel/gmake/Makefile
--- a/devel/gmake/Makefile
+++ b/devel/gmake/Makefile
@@ -47,6 +47,8 @@
 CONFIGURE_ARGS+=	ac_cv_func_strndup=no
 .endif
 
+CONFIGURE_ARGS+=	CFLAGS="-O2 -g -D__BEOS__"
+
 post-install:
 	${CHMOD} g-s ${DESTDIR}${PREFIX}/bin/${GNU_PROGRAM_PREFIX}make
 	${CHGRP} ${BINGRP} ${DESTDIR}${PREFIX}/bin/${GNU_PROGRAM_PREFIX}make
diff --git a/devel/imake/Makefile b/devel/imake/Makefile
--- a/devel/imake/Makefile
+++ b/devel/imake/Makefile
@@ -42,6 +42,9 @@
 SUBST_FILES.xmkmf=	xmkmf
 SUBST_SED.xmkmf=	-e "s/^configdirspec= */configdirspec=/"
 
+pre-build:
+	sed 's/-x c -/-x c - -x none/' -i ${WRKSRC}/imake.c
+
 .include "../../x11/xproto/buildlink3.mk"
 
 .include "../../mk/bsd.pkg.mk"
diff --git a/devel/libblkid/distinfo b/devel/libblkid/distinfo
--- a/devel/libblkid/distinfo
+++ b/devel/libblkid/distinfo
@@ -10,3 +10,4 @@
 SHA1 (patch-ae) = df77e8c6da425609309505fd447b66ade05426a3
 SHA1 (patch-af) = ba6e34b5635437e755192064bfc5865bfa9c913a
 SHA1 (patch-ag) = 1898774d311562938281f825b50a0e9aa44a447d
+SHA1 (patch-aq) = cb51811a855ad078e36113223c7afc3f19f5c32e
diff --git a/devel/libblkid/patches/patch-aq b/devel/libblkid/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/devel/libblkid/patches/patch-aq
@@ -0,0 +1,56 @@
+diff -r 4aa4d9dd485b shlibs/blkid/src/Makefile.am
+--- shlibs/blkid/src/Makefile.am	Sun May 06 03:11:57 2012 +0000
++++ shlibs/blkid/src/Makefile.am	Sun May 06 03:12:33 2012 +0000
+@@ -59,15 +59,15 @@
+ 
+ # move lib from $(usrlib_execdir) to $(libdir) if needed
+ install-exec-hook:
+-	if test "$(usrlib_execdir)" != "$(libdir)"; then \
+-		mkdir -p $(DESTDIR)$(libdir); \
+-		mv $(DESTDIR)$(usrlib_execdir)/libblkid.so.* $(DESTDIR)$(libdir); \
+-		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libblkid.so); \
+-		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
+-		(cd $(DESTDIR)$(usrlib_execdir) && \
+-			rm -f libblkid.so && \
+-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libblkid.so); \
+-	fi
++#	if test "$(usrlib_execdir)" != "$(libdir)"; then \
++#		mkdir -p $(DESTDIR)$(libdir); \
++#		mv $(DESTDIR)$(usrlib_execdir)/libblkid.so.* $(DESTDIR)$(libdir); \
++#		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libblkid.so); \
++#		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
++#		(cd $(DESTDIR)$(usrlib_execdir) && \
++#			rm -f libblkid.so && \
++#			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libblkid.so); \
++#	fi
+ 
+ uninstall-hook:
+ 	rm -f $(DESTDIR)$(libdir)/libblkid.so*
+diff -r 4aa4d9dd485b shlibs/blkid/src/Makefile.in
+--- shlibs/blkid/src/Makefile.in	Sun May 06 03:11:57 2012 +0000
++++ shlibs/blkid/src/Makefile.in	Sun May 06 03:12:33 2012 +0000
+@@ -922,15 +922,15 @@
+ 
+ # move lib from $(usrlib_execdir) to $(libdir) if needed
+ install-exec-hook:
+-	if test "$(usrlib_execdir)" != "$(libdir)"; then \
+-		mkdir -p $(DESTDIR)$(libdir); \
+-		mv $(DESTDIR)$(usrlib_execdir)/libblkid.so.* $(DESTDIR)$(libdir); \
+-		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libblkid.so); \
+-		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
+-		(cd $(DESTDIR)$(usrlib_execdir) && \
+-			rm -f libblkid.so && \
+-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libblkid.so); \
+-	fi
++#	if test "$(usrlib_execdir)" != "$(libdir)"; then \
++#		mkdir -p $(DESTDIR)$(libdir); \
++#		mv $(DESTDIR)$(usrlib_execdir)/libblkid.so.* $(DESTDIR)$(libdir); \
++#		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libblkid.so); \
++#		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
++#		(cd $(DESTDIR)$(usrlib_execdir) && \
++#			rm -f libblkid.so && \
++#			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libblkid.so); \
++#	fi
+ 
+ uninstall-hook:
+ 	rm -f $(DESTDIR)$(libdir)/libblkid.so*
diff --git a/devel/libuuid/distinfo b/devel/libuuid/distinfo
--- a/devel/libuuid/distinfo
+++ b/devel/libuuid/distinfo
@@ -4,3 +4,4 @@
 RMD160 (util-linux-2.19.1.tar.bz2) = e4763d21a3a32029ccc525a0222e271f0946a4fd
 Size (util-linux-2.19.1.tar.bz2) = 4396543 bytes
 SHA1 (patch-aa) = 07d66ef414f5781d75159b446d6c822b1e543477
+SHA1 (patch-aq) = f8ededcca54e2adb13f57e564623178f4d46ced6
diff --git a/devel/libuuid/patches/patch-aq b/devel/libuuid/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/devel/libuuid/patches/patch-aq
@@ -0,0 +1,56 @@
+diff -r f3d747a80d82 shlibs/uuid/src/Makefile.am
+--- shlibs/uuid/src/Makefile.am	Sun May 06 03:03:55 2012 +0000
++++ shlibs/uuid/src/Makefile.am	Sun May 06 03:04:28 2012 +0000
+@@ -25,15 +25,15 @@
+ 
+ # move lib from $(usrlib_execdir) to $(libdir) if needed
+ install-exec-hook:
+-	if test "$(usrlib_execdir)" != "$(libdir)"; then \
+-		mkdir -p $(DESTDIR)$(libdir); \
+-		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
+-		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
+-		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
+-		(cd $(DESTDIR)$(usrlib_execdir) && \
+-			rm -f libuuid.so && \
+-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
+-	fi
++#	if test "$(usrlib_execdir)" != "$(libdir)"; then \
++#		mkdir -p $(DESTDIR)$(libdir); \
++#		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
++#		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
++#		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
++#		(cd $(DESTDIR)$(usrlib_execdir) && \
++#			rm -f libuuid.so && \
++#			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
++#	fi
+ 
+ uninstall-hook:
+ 	rm -f $(DESTDIR)$(libdir)/libuuid.so*
+diff -r f3d747a80d82 shlibs/uuid/src/Makefile.in
+--- shlibs/uuid/src/Makefile.in	Sun May 06 03:03:55 2012 +0000
++++ shlibs/uuid/src/Makefile.in	Sun May 06 03:04:28 2012 +0000
+@@ -700,15 +700,15 @@
+ 
+ # move lib from $(usrlib_execdir) to $(libdir) if needed
+ install-exec-hook:
+-	if test "$(usrlib_execdir)" != "$(libdir)"; then \
+-		mkdir -p $(DESTDIR)$(libdir); \
+-		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
+-		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
+-		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
+-		(cd $(DESTDIR)$(usrlib_execdir) && \
+-			rm -f libuuid.so && \
+-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
+-	fi
++#	if test "$(usrlib_execdir)" != "$(libdir)"; then \
++#		mkdir -p $(DESTDIR)$(libdir); \
++#		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
++#		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
++#		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
++#		(cd $(DESTDIR)$(usrlib_execdir) && \
++#			rm -f libuuid.so && \
++#			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
++#	fi
+ 
+ uninstall-hook:
+ 	rm -f $(DESTDIR)$(libdir)/libuuid.so*
diff --git a/devel/m4/distinfo b/devel/m4/distinfo
--- a/devel/m4/distinfo
+++ b/devel/m4/distinfo
@@ -6,3 +6,4 @@
 SHA1 (patch-aa) = ef316620b49f78f46e9dea47032b0141814c1f43
 SHA1 (patch-ab) = b29840365e983623bce64e84ceab3504b8270b32
 SHA1 (patch-ac) = 7bd9164c659727a906ef54a2f1027e9fd19315ba
+SHA1 (patch-aq) = 361e3294fe76fd67087d4fb212fd16c79462a953
diff --git a/devel/m4/patches/patch-aq b/devel/m4/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/devel/m4/patches/patch-aq
@@ -0,0 +1,10 @@
+--- lib/freadahead.c
++++ lib/freadahead.c
+@@ -80,6 +80,6 @@
+   abort ();
+   return 0;
+ #else
+- #error "Please port gnulib freadahead.c to your platform! Look at the definition of fflush, fread, ungetc on your system, then report this to bug-gnulib."
++  return 1;
+ #endif
+ }
diff --git a/devel/ncurses/Makefile b/devel/ncurses/Makefile
--- a/devel/ncurses/Makefile
+++ b/devel/ncurses/Makefile
@@ -38,6 +38,9 @@
 
 PLIST_SRC=	${PKGDIR}/PLIST ${WRKDIR}/PLIST.terminfo
 
+# [Gregor] C++ build doesn't work
+CONFIGURE_ARGS+=	--without-cxx
+
 post-configure:
 	cd ${WRKSRC}/man;					\
 	for f in *.1m; do					\
diff --git a/devel/ncurses/Makefile.common b/devel/ncurses/Makefile.common
--- a/devel/ncurses/Makefile.common
+++ b/devel/ncurses/Makefile.common
@@ -16,14 +16,15 @@
 PKG_INSTALLATION_TYPES=	overwrite pkgviews
 PKG_DESTDIR_SUPPORT=	user-destdir
 
-USE_LANGUAGES=		c c++
+USE_LANGUAGES=		c
 USE_LIBTOOL=		yes
 USE_TOOLS+=		tbl
 GNU_CONFIGURE=		yes
 CONFIGURE_ARGS+=	--with-libtool
 CONFIGURE_ARGS+=	--enable-bsdpad
 CONFIGURE_ARGS+=	--enable-overwrite
-CONFIGURE_ARGS+=	--with-cxx-binding
+# [Gregor] No C++
+#CONFIGURE_ARGS+=	--with-cxx-binding
 CONFIGURE_ARGS+=	--without-ada
 CONFIGURE_ARGS+=	--without-gpm
 CONFIGURE_ARGS+=	--with-manpage-format=normal
diff --git a/devel/ncurses/PLIST b/devel/ncurses/PLIST
--- a/devel/ncurses/PLIST
+++ b/devel/ncurses/PLIST
@@ -11,14 +11,7 @@
 bin/tput
 bin/tset
 include/ncurses/curses.h
-include/ncurses/cursesapp.h
-include/ncurses/cursesf.h
-include/ncurses/cursesm.h
-include/ncurses/cursesp.h
-include/ncurses/cursesw.h
-include/ncurses/cursslk.h
 include/ncurses/eti.h
-include/ncurses/etip.h
 include/ncurses/form.h
 include/ncurses/menu.h
 include/ncurses/nc_tparm.h
@@ -32,7 +25,6 @@
 include/ncurses/unctrl.h
 lib/libform.la
 lib/libmenu.la
-lib/libncurses++.la
 lib/libncurses.la
 lib/libpanel.la
 man/man1/captoinfo.1
diff --git a/devel/nspr/Makefile b/devel/nspr/Makefile
--- a/devel/nspr/Makefile
+++ b/devel/nspr/Makefile
@@ -70,6 +70,13 @@
 BUILDLINK_PASSTHRU_DIRS+=	Developer/Headers/FlatCarbon
 .endif
 
+# [Gregor] SOL_TCP is deprecated
+CFLAGS+=-DSOL_TCP=IPPROTO_TCP
+
+# [Gregor] the detection is broken
+pre-build:
+	${SED} 's/#define _PR_HAVE_GETPROTO_R.*//' -i ${WRKSRC}/nsprpub/pr/src/misc/prnetdb.c
+
 pre-configure:
 # sanity check: make sure the nspr release is what we think it is.
 #	@set -e; if [ '"${NSPR_RELEASE}"' !=			\
diff --git a/devel/nss/Makefile b/devel/nss/Makefile
--- a/devel/nss/Makefile
+++ b/devel/nss/Makefile
@@ -48,6 +48,17 @@
 MAKE_ENV+=	NSS_USE_SYSTEM_SQLITE=1
 MAKE_ENV+=	EXTRA_SHARED_LIBS="-Wl,-R${BUILDLINK_PREFIX.nspr}/lib/nspr -Wl,-R,${PREFIX}/lib/nss -L${BUILDLINK_PREFIX.nspr}/lib/nspr -lplc4 -lplds4 -lnspr4"
 
+# [Gregor] And just where is this type SUPPOSED to come from?
+_WRAP_EXTRA_ARGS.CC+=-Duint=uint32_t
+
+# [Gregor] This build loses, but needs, our CFLAGS
+_WRAP_EXTRA_ARGS.CC+=-D_GNU_SOURCE -D_BSD_SOURCE
+
+# [Gregor] More misdetection
+pre-build:
+	${SED} 's/DHAVE_SYS_CDEFS_H/UHAVE_SYS_CDEFS_H/' -i ${WRKSRC}/security/dbm/config/config.mk
+	${SED} 's/(ffs/(__builtin_ffs/g' -i ${WRKSRC}/security/nss/cmd/checkcert/checkcert.c
+
 pre-configure:
 # sanity check: make sure the nss release is what we think it is.
 #	@set -e; if [ '"${NSS_RELEASE}"' !=			\
diff --git a/devel/pcre/Makefile b/devel/pcre/Makefile
--- a/devel/pcre/Makefile
+++ b/devel/pcre/Makefile
@@ -21,6 +21,8 @@
 CONFIGURE_ARGS+=	--enable-utf8 --enable-unicode-properties
 CONFIGURE_ARGS+=	--enable-pcre16
 
+CONFIGURE_ARGS+=	--disable-cpp
+
 SUBST_CLASSES+=		rpath
 SUBST_STAGE.rpath=	pre-configure
 SUBST_MESSAGE.rpath=	Fixing rpath output in pcre-config
diff --git a/devel/pcre/PLIST b/devel/pcre/PLIST
--- a/devel/pcre/PLIST
+++ b/devel/pcre/PLIST
@@ -3,18 +3,12 @@
 bin/pcregrep
 bin/pcretest
 include/pcre.h
-include/pcre_scanner.h
-include/pcre_stringpiece.h
-include/pcrecpp.h
-include/pcrecpparg.h
 include/pcreposix.h
 lib/libpcre.la
 lib/libpcre16.la
-lib/libpcrecpp.la
 lib/libpcreposix.la
 lib/pkgconfig/libpcre.pc
 lib/pkgconfig/libpcre16.pc
-lib/pkgconfig/libpcrecpp.pc
 lib/pkgconfig/libpcreposix.pc
 man/man1/pcre-config.1
 man/man1/pcregrep.1
@@ -75,7 +69,6 @@
 man/man3/pcrebuild.3
 man/man3/pcrecallout.3
 man/man3/pcrecompat.3
-man/man3/pcrecpp.3
 man/man3/pcrejit.3
 man/man3/pcrelimits.3
 man/man3/pcrematching.3
@@ -127,7 +120,6 @@
 share/doc/pcre/html/pcrebuild.html
 share/doc/pcre/html/pcrecallout.html
 share/doc/pcre/html/pcrecompat.html
-share/doc/pcre/html/pcrecpp.html
 share/doc/pcre/html/pcredemo.html
 share/doc/pcre/html/pcregrep.html
 share/doc/pcre/html/pcrejit.html
diff --git a/devel/xulrunner/Makefile b/devel/xulrunner/Makefile
--- a/devel/xulrunner/Makefile
+++ b/devel/xulrunner/Makefile
@@ -34,6 +34,9 @@
 .include "mozilla-common.mk"
 .include "options.mk"
 
+# [Gregor] To make some patches work
+CFLAGS+=		-D__musl__
+
 pre-configure:
 	cd ${WRKSRC} && autoconf
 	cd ${WRKSRC}/js/src && autoconf
diff --git a/devel/xulrunner/distinfo b/devel/xulrunner/distinfo
--- a/devel/xulrunner/distinfo
+++ b/devel/xulrunner/distinfo
@@ -106,3 +106,4 @@
 SHA1 (patch-xo) = 437cbc7e0ed8777ae2313165f8754759d8eee01d
 SHA1 (patch-xpcom_Makefile.in) = ad9c0ed4347b0ed4f51a6f0f34ac48c65d2152c8
 SHA1 (patch-xpcom_idl-parser_header.py) = cd14f4056a31f15066aebb85e5c14fba2aa37f13
+SHA1 (patch-zq) = b5705937db9663ec56962e25e1f007182174edb8
diff --git a/devel/xulrunner/patches/patch-zq b/devel/xulrunner/patches/patch-zq
new file mode 100644
--- /dev/null
+++ b/devel/xulrunner/patches/patch-zq
@@ -0,0 +1,450 @@
+diff -r 39c8c27f8b14 build/unix/elfhack/elfhack.cpp
+--- build/unix/elfhack/elfhack.cpp	Tue May 08 18:47:38 2012 +0000
++++ build/unix/elfhack/elfhack.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -38,6 +38,7 @@
+ #undef NDEBUG
+ #include <assert.h>
+ #include <cstring>
++#include <strings.h>
+ #include <cstdlib>
+ #include <cstdio>
+ #include "elfxx.h"
+diff -r 39c8c27f8b14 ipc/chromium/src/base/debug_util_posix.cc
+--- ipc/chromium/src/base/debug_util_posix.cc	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/debug_util_posix.cc	Thu May 10 14:14:21 2012 +0000
+@@ -5,7 +5,7 @@
+ #include "build/build_config.h"
+ #include "base/debug_util.h"
+ 
+-#define MOZ_HAVE_EXECINFO_H (defined(__linux__) && !defined(ANDROID))
++#define MOZ_HAVE_EXECINFO_H (defined(__linux__) && !defined(ANDROID) && !defined(__musl__))
+ 
+ #include <errno.h>
+ #include <fcntl.h>
+diff -r 39c8c27f8b14 ipc/chromium/src/base/file_util.cc
+--- ipc/chromium/src/base/file_util.cc	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/file_util.cc	Thu May 10 14:14:21 2012 +0000
+@@ -8,7 +8,7 @@
+ #include <io.h>
+ #endif
+ #include <stdio.h>
+-#if defined(ANDROID)
++#if defined(ANDROID) || defined(__musl__)
+ #include <unistd.h>
+ #endif
+ 
+diff -r 39c8c27f8b14 ipc/chromium/src/base/file_util.h
+--- ipc/chromium/src/base/file_util.h	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/file_util.h	Thu May 10 14:14:21 2012 +0000
+@@ -12,7 +12,7 @@
+ 
+ #if defined(OS_WIN)
+ #include <windows.h>
+-#elif defined(ANDROID)
++#elif defined(ANDROID) || defined(__musl__)
+ #include <sys/stat.h>
+ #elif defined(OS_POSIX) 
+ #include <sys/types.h>
+@@ -466,7 +466,7 @@
+ #if defined(OS_WIN)
+   WIN32_FIND_DATA find_data_;
+   HANDLE find_handle_;
+-#elif defined(ANDROID)
++#elif defined(ANDROID) || defined(__musl__)
+   void *fts_;
+ #elif defined(OS_POSIX)
+   FTS* fts_;
+diff -r 39c8c27f8b14 ipc/chromium/src/base/file_util_linux.cc
+--- ipc/chromium/src/base/file_util_linux.cc	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/file_util_linux.cc	Thu May 10 14:14:21 2012 +0000
+@@ -5,6 +5,7 @@
+ #include "base/file_util.h"
+ 
+ #include <fcntl.h>
++#include <unistd.h>
+ 
+ #include <string>
+ #include <vector>
+diff -r 39c8c27f8b14 ipc/chromium/src/base/file_util_posix.cc
+--- ipc/chromium/src/base/file_util_posix.cc	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/file_util_posix.cc	Thu May 10 14:14:21 2012 +0000
+@@ -8,13 +8,15 @@
+ #include <errno.h>
+ #include <fcntl.h>
+ #include <fnmatch.h>
+-#ifndef ANDROID
++#if !defined(ANDROID) && !defined(__musl__)
+ #include <fts.h>
+ #endif
+ #include <libgen.h>
+ #include <stdio.h>
+ #include <string.h>
++#ifndef __musl__
+ #include <sys/errno.h>
++#endif
+ #include <sys/mman.h>
+ #include <sys/stat.h>
+ #include <sys/types.h>
+@@ -31,7 +33,7 @@
+ #include "base/time.h"
+ 
+ // FreeBSD/OpenBSD lacks stat64, but its stat handles files >2GB just fine
+-#ifndef __linux__
++#if !defined(__linux__) || defined(__musl__)
+ #define stat64 stat
+ #endif
+ 
+@@ -123,7 +125,7 @@
+   if (!recursive)
+     return (rmdir(path_str) == 0);
+ 
+-#ifdef ANDROID
++#if defined(ANDROID) || defined(__musl__)
+   // XXX Need ftsless impl for bionic
+   return false;
+ #else
+@@ -196,7 +198,7 @@
+     return false;
+   }
+ 
+-#ifdef ANDROID
++#if defined(ANDROID) || defined(__musl__)
+   // XXX Need ftsless impl for bionic
+   return false;
+ #else
+@@ -419,7 +421,7 @@
+   std::string tmpdir_string = tmpdir.value();
+   // this should be OK since mkdtemp just replaces characters in place
+   char* buffer = const_cast<char*>(tmpdir_string.c_str());
+-#ifdef ANDROID
++#if defined(ANDROID) || defined(__musl__)
+   char* dtemp = NULL;
+ #else
+   char* dtemp = mkdtemp(buffer);
+@@ -549,7 +551,7 @@
+ }
+ 
+ FileEnumerator::~FileEnumerator() {
+-#ifndef ANDROID
++#if !defined(ANDROID) && !defined(__musl__)
+   if (fts_)
+     fts_close(fts_);
+ #endif
+@@ -561,7 +563,7 @@
+   if (!is_in_find_op_)
+     return;
+ 
+-#ifndef ANDROID
++#if !defined(ANDROID) && !defined(__musl__)
+   memcpy(&(info->stat), fts_ent_->fts_statp, sizeof(info->stat));
+   info->filename.assign(fts_ent_->fts_name);
+ #endif
+@@ -572,7 +574,7 @@
+ // large directories with many files this can be quite deep.
+ // TODO(erikkay) - get rid of this recursive pattern
+ FilePath FileEnumerator::Next() {
+-#ifdef ANDROID
++#if defined(ANDROID) || defined(__musl__)
+   return FilePath();
+ #else
+   if (!is_in_find_op_) {
+diff -r 39c8c27f8b14 ipc/chromium/src/base/message_pump_libevent.cc
+--- ipc/chromium/src/base/message_pump_libevent.cc	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/message_pump_libevent.cc	Thu May 10 14:14:21 2012 +0000
+@@ -6,6 +6,7 @@
+ 
+ #include <errno.h>
+ #include <fcntl.h>
++#include <unistd.h>
+ 
+ #include "eintr_wrapper.h"
+ #include "base/logging.h"
+diff -r 39c8c27f8b14 ipc/chromium/src/base/process_util.h
+--- ipc/chromium/src/base/process_util.h	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/process_util.h	Thu May 10 14:14:21 2012 +0000
+@@ -16,6 +16,7 @@
+ #elif defined(OS_LINUX)
+ #include <limits.h>
+ #include <sys/types.h>
++#include <dirent.h>
+ #elif defined(OS_MACOSX)
+ #include <mach/mach.h>
+ #endif
+diff -r 39c8c27f8b14 ipc/chromium/src/base/third_party/nspr/prtime.cc
+--- ipc/chromium/src/base/third_party/nspr/prtime.cc	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/third_party/nspr/prtime.cc	Thu May 10 14:14:21 2012 +0000
+@@ -173,7 +173,7 @@
+     result *= kSecondsToMicroseconds;
+     result += exploded->tm_usec;
+     return result;
+-#elif defined(ANDROID)
++#elif defined(ANDROID) || defined(__musl__)
+ #define LL_ADD(r, a, b)     ((r) = (a) + (b))
+ #define LL_SUB(r, a, b)     ((r) = (a) - (b))
+ 
+diff -r 39c8c27f8b14 ipc/chromium/src/base/time_posix.cc
+--- ipc/chromium/src/base/time_posix.cc	Tue May 08 18:47:38 2012 +0000
++++ ipc/chromium/src/base/time_posix.cc	Thu May 10 14:14:21 2012 +0000
+@@ -13,6 +13,9 @@
+ #else
+ #include <time.h>
+ #endif
++#if defined(__musl__)
++#include <unistd.h>
++#endif
+ 
+ #include <limits>
+ 
+@@ -62,11 +65,13 @@
+   timestruct.tm_wday   = exploded.day_of_week;  // mktime/timegm ignore this
+   timestruct.tm_yday   = 0;     // mktime/timegm ignore this
+   timestruct.tm_isdst  = -1;    // attempt to figure it out
++#ifndef __musl__
+   timestruct.tm_gmtoff = 0;     // not a POSIX field, so mktime/timegm ignore
+   timestruct.tm_zone   = NULL;  // not a POSIX field, so mktime/timegm ignore
++#endif  
+ 
+   time_t seconds;
+-#ifdef ANDROID
++#if defined(ANDROID) || defined(__musl__)
+     seconds = mktime(&timestruct);
+ #else
+   if (is_local)
+diff -r 39c8c27f8b14 js/src/ctypes/CTypes.cpp
+--- js/src/ctypes/CTypes.cpp	Tue May 08 18:47:38 2012 +0000
++++ js/src/ctypes/CTypes.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -490,7 +490,7 @@
+ #ifdef WIN32
+   return _finite(f) != 0;
+ #else
+-  return finite(f);
++  return isfinite(f);
+ #endif
+ }
+ 
+diff -r 39c8c27f8b14 js/src/jsnativestack.cpp
+--- js/src/jsnativestack.cpp	Tue May 08 18:47:38 2012 +0000
++++ js/src/jsnativestack.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -37,6 +37,9 @@
+  * ***** END LICENSE BLOCK ***** */
+ 
+ #include <stdlib.h>
++#ifdef __musl__
++#include <stdio.h>
++#endif
+ #include "jstypes.h"
+ #include "jsnativestack.h"
+ 
+@@ -137,6 +140,42 @@
+     return ptib->tib_pstacklimit;
+ }
+ 
++#elif defined(__musl__)
++
++void *
++GetNativeStackBaseImpl()
++{
++    long onstack;
++    FILE *procselfmaps;
++    char *line = NULL;
++    size_t lineSz = 0;
++    
++    onstack = (long) (void *) &onstack;
++    
++    procselfmaps = fopen("/proc/self/maps", "r");
++    if (procselfmaps == NULL) abort();
++    
++    while (getdelim(&line, &lineSz, '\n', procselfmaps) > 0) {
++        long from, to;
++        if (sscanf(line, "%lx-%lx", &from, &to) != 2) continue;
++        if (from <= onstack && onstack < to) {
++            /* found our stack */
++            free(line);
++            fclose(procselfmaps);
++#if JS_STACK_GROWTH_DIRECTION > 0
++            return (void *) from;
++#else
++            return (void *) to;
++#endif
++        }
++    }
++    free(line);
++    fclose(procselfmaps);
++    
++    /* didn't find it! */
++    abort();
++}
++
+ #else /* XP_UNIX */
+ 
+ void *
+diff -r 39c8c27f8b14 memory/mozalloc/mozalloc.cpp
+--- memory/mozalloc/mozalloc.cpp	Tue May 08 18:47:38 2012 +0000
++++ memory/mozalloc/mozalloc.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -259,7 +259,7 @@
+ 
+ #if defined(XP_MACOSX)
+     return malloc_size(ptr);
+-#elif defined(MOZ_MEMORY) || defined(XP_LINUX)
++#elif (defined(MOZ_MEMORY) || defined(XP_LINUX)) && !defined(__musl__)
+     // XXX: the |defined(XP_LINUX)| may be too lax;  some Linux installations
+     // might use a libc that doesn't have malloc_usable_size.  Let's fix this
+     // if/when it happens.
+diff -r 39c8c27f8b14 toolkit/library/Makefile.in
+--- toolkit/library/Makefile.in	Tue May 08 18:47:38 2012 +0000
++++ toolkit/library/Makefile.in	Thu May 10 14:14:21 2012 +0000
+@@ -561,8 +561,8 @@
+ 
+ ifeq ($(OS_ARCH),FreeBSD)
+ OS_LIBS += $(call EXPAND_LIBNAME,kvm)
++endif
+ EXTRA_DSO_LDOPTS += -lplc4 -lplds4 -lnspr4 -lnssutil3
+-endif
+ 
+ ifeq ($(OS_ARCH),NetBSD)
+ OS_LIBS += $(call EXPAND_LIBNAME,kvm)
+diff -r 39c8c27f8b14 toolkit/mozapps/update/updater/updater.cpp
+--- toolkit/mozapps/update/updater/updater.cpp	Tue May 08 18:47:38 2012 +0000
++++ toolkit/mozapps/update/updater/updater.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -127,7 +127,9 @@
+ #else
+ # include <sys/wait.h>
+ # include <unistd.h>
++#ifndef __musl__
+ # include <fts.h>
++#endif
+ 
+ #ifdef XP_MACOSX
+ # include <sys/time.h>
+@@ -2238,6 +2240,7 @@
+ 
+ int add_dir_entries(const NS_tchar *dirpath, ActionList *list)
+ {
++#ifndef __musl__
+   int rv = OK;
+   FTS *ftsdir;
+   FTSENT *ftsdirEntry;
+@@ -2343,6 +2346,9 @@
+   fts_close(ftsdir);
+ 
+   return rv;
++#else
++  return -1;
++#endif
+ }
+ #endif
+ 
+diff -r 39c8c27f8b14 toolkit/xre/nsSigHandlers.cpp
+--- toolkit/xre/nsSigHandlers.cpp	Tue May 08 18:47:38 2012 +0000
++++ toolkit/xre/nsSigHandlers.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -60,7 +60,7 @@
+ #include <sys/resource.h>
+ #include <unistd.h>
+ #include <stdlib.h> // atoi
+-#ifndef ANDROID // no Android impl
++#if !defined(ANDROID) && !defined(__musl__) // no Android impl
+ #  include <ucontext.h>
+ #endif
+ #endif
+@@ -190,7 +190,7 @@
+   *mxcsr &= ~SSE_STATUS_FLAGS; /* clear all pending SSE exceptions */
+ #endif
+ #endif
+-#if defined(LINUX) && !defined(ANDROID)
++#if defined(LINUX) && !defined(ANDROID) && !defined(__musl__)
+   ucontext_t *uc = (ucontext_t *)context;
+ 
+ #if defined(__i386__)
+diff -r 39c8c27f8b14 tools/profiler/sps/platform-linux.cc
+--- tools/profiler/sps/platform-linux.cc	Tue May 08 18:47:38 2012 +0000
++++ tools/profiler/sps/platform-linux.cc	Thu May 10 14:14:21 2012 +0000
+@@ -47,7 +47,7 @@
+ #define PATH_MAX_TOSTRING(x) #x
+ #define PATH_MAX_STRING(x) PATH_MAX_TOSTRING(x)
+ 
+-#if defined(__GLIBC__)
++#if defined(__GLIBC__) || defined(__musl__)
+ // glibc doesn't implement gettid(2).
+ #include <sys/syscall.h>
+ pid_t gettid()
+@@ -56,6 +56,10 @@
+ }
+ #endif
+ 
++#if defined(__musl__)
++typedef long useconds_t;
++#endif
++
+ #ifdef ENABLE_SPS_LEAF_DATA
+ /* a crapy version of getline, because it's not included in bionic */
+ static ssize_t getline(char **lineptr, size_t *n, FILE *stream)
+diff -r 39c8c27f8b14 xpcom/ds/nsMathUtils.h
+--- xpcom/ds/nsMathUtils.h	Tue May 08 18:47:38 2012 +0000
++++ xpcom/ds/nsMathUtils.h	Thu May 10 14:14:21 2012 +0000
+@@ -134,6 +134,12 @@
+     // Darwin has deprecated |finite| and recommends |isfinite|. The former is
+     // not present in the iOS SDK.
+     return isfinite(d);
++#elif defined(__musl__)
++#if defined(__cplusplus) && !defined(isfinite)
++    return std::isfinite(d);
++#else
++    return isfinite(d);
++#endif
+ #else
+     return finite(d);
+ #endif
+diff -r 39c8c27f8b14 xpcom/glue/standalone/nsGlueLinkingDlopen.cpp
+--- xpcom/glue/standalone/nsGlueLinkingDlopen.cpp	Tue May 08 18:47:38 2012 +0000
++++ xpcom/glue/standalone/nsGlueLinkingDlopen.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -41,7 +41,7 @@
+ #include "nsXPCOMGlue.h"
+ #include "nscore.h"
+ 
+-#if defined(LINUX) && !defined(ANDROID)
++#if defined(LINUX) && !defined(ANDROID) && !defined(__musl__)
+ #define _GNU_SOURCE 
+ #include <fcntl.h>
+ #include <unistd.h>
+@@ -159,7 +159,7 @@
+     sTop = d;
+ }
+ 
+-#if defined(LINUX) && !defined(ANDROID)
++#if defined(LINUX) && !defined(ANDROID) && !defined(__musl__)
+ static const unsigned int bufsize = 4096;
+ 
+ #ifdef HAVE_64BIT_OS
+@@ -218,7 +218,7 @@
+ static void
+ ReadDependentCB(const char *aDependentLib, bool do_preload)
+ {
+-#if defined(LINUX) && !defined(ANDROID)
++#if defined(LINUX) && !defined(ANDROID) && !defined(__musl__)
+     if (do_preload)
+         preload(aDependentLib);
+ #endif
+diff -r 39c8c27f8b14 xpcom/io/nsLocalFileUnix.cpp
+--- xpcom/io/nsLocalFileUnix.cpp	Tue May 08 18:47:38 2012 +0000
++++ xpcom/io/nsLocalFileUnix.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -436,7 +436,7 @@
+         PR_Delete(mPath.get());
+     }
+ 
+-#if defined(LINUX) && !defined(ANDROID)
++#if defined(LINUX) && !defined(ANDROID) && !defined(__musl__)
+     if (flags & OS_READAHEAD) {
+         readahead(PR_FileDesc2NativeHandle(*_retval), 0, 0);
+     }
+diff -r 39c8c27f8b14 xpcom/threads/nsThread.cpp
+--- xpcom/threads/nsThread.cpp	Tue May 08 18:47:38 2012 +0000
++++ xpcom/threads/nsThread.cpp	Thu May 10 14:14:21 2012 +0000
+@@ -53,7 +53,7 @@
+                       _XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) &&           \
+                       !(_POSIX_C_SOURCE >= 200809L || _XOPEN_SOURCE >= 700)
+ 
+-#if defined(XP_UNIX) && !defined(ANDROID) && !defined(DEBUG) && HAVE_UALARM \
++#if defined(XP_UNIX) && !defined(ANDROID) && !defined(__musl__) && !defined(DEBUG) && HAVE_UALARM \
+   && defined(_GNU_SOURCE)
+ # define MOZ_CANARY
+ # include <unistd.h>
diff --git a/editors/vim-share/Makefile.common b/editors/vim-share/Makefile.common
--- a/editors/vim-share/Makefile.common
+++ b/editors/vim-share/Makefile.common
@@ -86,5 +86,8 @@
 CONFIGURE_ARGS+=	--enable-rubyinterp
 .endif
 
+# [Gregor] vim doesn't recognize that it needs curses to build ANYTHING, even the share component
+.include "../../mk/curses.buildlink3.mk"
+
 post-patch:
 	find ${WRKDIR} -name '*.orig_dist' -exec rm -f {} \;
diff --git a/filesystems/fuse/distinfo b/filesystems/fuse/distinfo
--- a/filesystems/fuse/distinfo
+++ b/filesystems/fuse/distinfo
@@ -16,3 +16,4 @@
 SHA1 (patch-am) = 2abd07b403896bff5aad6cd865cb473553851a22
 SHA1 (patch-an) = 3f119dae5f08996dd483a0fb4fa0365507be2070
 SHA1 (patch-util_fusermount.c) = c65c377b917dc0c34cf5b8fb4d70e75075debcf9
+SHA1 (patch-qq) = f8dc23f988fa537dcb41bdc54f4dc550eb479914
diff --git a/filesystems/fuse/patches/patch-qq b/filesystems/fuse/patches/patch-qq
new file mode 100644
--- /dev/null
+++ b/filesystems/fuse/patches/patch-qq
@@ -0,0 +1,48 @@
+diff -r 18a7de07ebb3 example/fioclient.c
+--- example/fioclient.c	Tue May 08 17:27:11 2012 +0000
++++ example/fioclient.c	Tue May 08 17:34:04 2012 +0000
+@@ -10,7 +10,7 @@
+ */
+ 
+ #include <sys/types.h>
+-#include <sys/fcntl.h>
++#include <fcntl.h>
+ #include <sys/stat.h>
+ #include <sys/ioctl.h>
+ #include <stdio.h>
+diff -r 18a7de07ebb3 lib/mount_util.c
+--- lib/mount_util.c	Tue May 08 17:27:11 2012 +0000
++++ lib/mount_util.c	Tue May 08 17:34:04 2012 +0000
+@@ -24,6 +24,12 @@
+ #include <sys/mount.h>
+ #include <sys/param.h>
+ 
++#ifndef _PATH_MOUNTED
++#ifdef MOUNTED
++#define _PATH_MOUNTED MOUNTED
++#endif
++#endif
++
+ #ifdef __NetBSD__
+ #define umount2(mnt, flags) unmount(mnt, (flags == 2) ? MNT_FORCE : 0)
+ #define mtab_needs_update(mnt) 0
+diff -r 18a7de07ebb3 util/fusermount.c
+--- util/fusermount.c	Tue May 08 17:27:11 2012 +0000
++++ util/fusermount.c	Tue May 08 17:34:04 2012 +0000
+@@ -29,6 +29,16 @@
+ #include <sys/utsname.h>
+ #include <sched.h>
+ 
++#ifndef _PATH_MOUNTED
++#ifdef MOUNTED
++#define _PATH_MOUNTED MOUNTED
++#endif
++#endif
++
++#ifndef __WCLONE
++#define __WCLONE      0x80000000
++#endif
++
+ #define FUSE_COMMFD_ENV		"_FUSE_COMMFD"
+ 
+ #define FUSE_DEV_OLD "/proc/fs/fuse/dev"
diff --git a/graphics/MesaLib/Makefile b/graphics/MesaLib/Makefile
--- a/graphics/MesaLib/Makefile
+++ b/graphics/MesaLib/Makefile
@@ -72,6 +72,7 @@
 
 pre-build:
 	cd ${WRKSRC} && ${RM} -fr src/glu src/glut progs
+	sed -i 's/#if defined(__linux__) && defined(__i386__)/#if 0/' ${WRKSRC}/src/mesa/main/glheader.h
 
 .include "../../mk/compiler.mk"
 
diff --git a/graphics/aalib/distinfo b/graphics/aalib/distinfo
--- a/graphics/aalib/distinfo
+++ b/graphics/aalib/distinfo
@@ -6,3 +6,4 @@
 SHA1 (patch-aa) = 927d69a8f499b99f0bd5f13cf220ce8cbb7ea591
 SHA1 (patch-ab) = 272f6f10dbe7a525c582a9b4c865f0ab9b2f9fba
 SHA1 (patch-ac) = 8d9c812b9b871f44dd1285d6a635eeab6e49bf49
+SHA1 (patch-aq) = 285c28a811719d0e83f8df29c3a6bf93fb3eef8f
diff --git a/graphics/aalib/patches/patch-aq b/graphics/aalib/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/graphics/aalib/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 6fe3c134e687 src/aalinuxkbd.c
+--- src/aalinuxkbd.c	Mon May 07 17:51:38 2012 +0000
++++ src/aalinuxkbd.c	Mon May 07 17:51:56 2012 +0000
+@@ -346,7 +346,7 @@
+ #endif
+ static char sig2catch[] =
+ {SIGHUP, SIGINT, SIGQUIT, SIGILL,
+- SIGTRAP, SIGIOT, SIGBUS, SIGFPE,
++ SIGTRAP, SIGABRT, SIGBUS, SIGFPE,
+  SIGSEGV, SIGPIPE, SIGALRM, SIGTERM,
+  SIGXCPU, SIGXFSZ, SIGVTALRM,
+ 	   /* SIGPROF , */ SIGPWR};
diff --git a/graphics/cairo/Makefile b/graphics/cairo/Makefile
--- a/graphics/cairo/Makefile
+++ b/graphics/cairo/Makefile
@@ -38,6 +38,8 @@
 # For snprintf() and ctime_r()
 CPPFLAGS.SunOS+=		-D__EXTENSIONS__ -D_POSIX_PTHREAD_SEMANTICS
 
+CPPFLAGS+=		-Dffs=__builtin_ffs
+
 TEST_TARGET=			check
 
 # PR#43928
diff --git a/graphics/gdk-pixbuf2/PLIST b/graphics/gdk-pixbuf2/PLIST
--- a/graphics/gdk-pixbuf2/PLIST
+++ b/graphics/gdk-pixbuf2/PLIST
@@ -72,104 +72,3 @@
 share/gtk-doc/html/gdk-pixbuf/rn02.html
 share/gtk-doc/html/gdk-pixbuf/style.css
 share/gtk-doc/html/gdk-pixbuf/up.png
-share/locale/af/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ang/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ar/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/as/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ast/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/az/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/be/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/be@latin/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/bg/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/bn/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/bn_IN/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/br/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/bs/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ca/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ca@valencia/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/crh/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/cs/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/cy/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/da/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/de/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/dz/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/el/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/en@shaw/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/en_CA/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/en_GB/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/eo/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/es/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/et/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/eu/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/fa/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/fi/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/fr/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ga/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/gl/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/gu/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/he/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/hi/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/hr/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/hu/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/hy/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ia/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/id/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/io/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/is/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/it/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ja/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ka/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/kn/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ko/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ku/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/li/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/lt/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/lv/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/mai/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/mi/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/mk/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ml/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/mn/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/mr/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ms/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/my/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/nb/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/nds/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ne/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/nl/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/nn/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/nso/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/oc/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/or/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/pa/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/pl/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ps/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/pt/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/pt_BR/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ro/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ru/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/si/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/sk/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/sl/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/sq/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/sr/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/sr@ije/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/sr@latin/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/sv/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ta/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/te/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/th/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/tk/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/tr/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/tt/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/ug/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/uk/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/uz/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/uz@cyrillic/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/vi/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/wa/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/xh/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/yi/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/zh_CN/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/zh_HK/LC_MESSAGES/gdk-pixbuf.mo
-share/locale/zh_TW/LC_MESSAGES/gdk-pixbuf.mo
diff --git a/graphics/gimp/PLIST b/graphics/gimp/PLIST
--- a/graphics/gimp/PLIST
+++ b/graphics/gimp/PLIST
@@ -307,7 +307,6 @@
 lib/gimp/2.0/plug-ins/file-tiff-load
 lib/gimp/2.0/plug-ins/file-tiff-save
 lib/gimp/2.0/plug-ins/file-uri
-lib/gimp/2.0/plug-ins/file-wmf
 lib/gimp/2.0/plug-ins/file-xbm
 lib/gimp/2.0/plug-ins/file-xjt
 ${PLIST.remote}lib/gimp/2.0/plug-ins/file-xpm
diff --git a/graphics/jpeg/buildlink3.mk b/graphics/jpeg/buildlink3.mk
--- a/graphics/jpeg/buildlink3.mk
+++ b/graphics/jpeg/buildlink3.mk
@@ -5,9 +5,10 @@
 .if !defined(JPEG_BUILDLINK3_MK)
 JPEG_BUILDLINK3_MK:=
 
-BUILDLINK_API_DEPENDS.jpeg+=	jpeg>=8nb1
-#BUILDLINK_ABI_DEPENDS.jpeg+=	jpeg>=8nb1
-BUILDLINK_ABI_DEPENDS.jpeg?=	jpeg>=8nb1
+# [Gregor] Reduced to 8 due to version diffs between pkgresolve and pkgsrc
+BUILDLINK_API_DEPENDS.jpeg+=	jpeg>=8
+#BUILDLINK_ABI_DEPENDS.jpeg+=	jpeg>=8
+BUILDLINK_ABI_DEPENDS.jpeg?=	jpeg>=8
 BUILDLINK_PKGSRCDIR.jpeg?=	../../graphics/jpeg
 .endif # JPEG_BUILDLINK3_MK
 
diff --git a/lang/f2c/distinfo b/lang/f2c/distinfo
--- a/lang/f2c/distinfo
+++ b/lang/f2c/distinfo
@@ -6,3 +6,4 @@
 SHA1 (patch-aa) = f1224ff17cdc96f083a247dce88b690e8b939eb0
 SHA1 (patch-ad) = 7f204a45891e3632589ea8fb0e4e379a0a792dea
 SHA1 (patch-ak) = 8652a54df300ddd60d9b1a9594d312349b782cf4
+SHA1 (patch-aq) = dd5d4e79bd1d61d7ce152b0934667ba4a1888692
diff --git a/lang/f2c/patches/patch-aq b/lang/f2c/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/lang/f2c/patches/patch-aq
@@ -0,0 +1,13 @@
+diff -r 6bba7eaf719c src/sysdep.c
+--- src/sysdep.c	Sat Apr 28 14:46:46 2012 +0000
++++ src/sysdep.c	Sat Apr 28 14:47:22 2012 +0000
+@@ -83,6 +83,9 @@
+ #include <sys/wait.h>
+ #include <unistd.h>
+ 
++#include <sys/types.h>
++#include <sys/stat.h>
++
+  void
+ #ifdef KR_headers
+ Un_link_all(cdelete)
diff --git a/lang/perl5/Makefile b/lang/perl5/Makefile
--- a/lang/perl5/Makefile
+++ b/lang/perl5/Makefile
@@ -50,6 +50,7 @@
 CONFIGURE_ARGS+=	-sde
 CONFIGURE_ARGS+=	-D ccflags=${CFLAGS:Q}
 CONFIGURE_ARGS+=	-D ldflags=${LDFLAGS:Q}
+CONFIGURE_ARGS+=        -D inc_version_list=none
 MAKE_ENV+=		LC_ALL="C"
 
 TEST_TARGET=		test
@@ -320,6 +321,10 @@
 
 pre-configure:
 	cd ${WRKSRC} && find `pwd` -name "*.orig" -type f -exec ${RM} -f {} \;
+	sed 's/-fstack-protector/-fno-stack-protector/g' -i ${WRKSRC}/Configure
+	
+pre-build:	
+	sed 's/define HAS_CRYPT_R/undef HAS_CRYPT_R/g' -i ${WRKSRC}/config.h
 
 post-build:
 	${SED}	-e "s,@PERL5@,"${PERL5:Q}",g"				\
diff --git a/lang/python26/Makefile b/lang/python26/Makefile
--- a/lang/python26/Makefile
+++ b/lang/python26/Makefile
@@ -118,22 +118,7 @@
 .endif
 
 PLIST_VARS+=	dll nis no-nis
-.if ${OPSYS} == "IRIX"
-.  if ${ABI} == "64"
-PLIST.no-nis=	yes
-.  else
-PLIST.nis=	yes
-.  endif
-.else
 PLIST.dll=	yes
-.  if ${OPSYS} == "MirBSD"
-# neither nis nor no-nis
-.  elif ${OPSYS} != "NetBSD" || exists(/usr/bin/ypcat)
-PLIST.nis=	yes
-.  else
-PLIST.no-nis=	yes
-.  endif
-.endif
 
 .include "../../mk/bdb.buildlink3.mk"
 MAKE_ENV+=	PY_BDB_TYPE=${BDB_TYPE}
diff --git a/lang/python27/PLIST.common b/lang/python27/PLIST.common
--- a/lang/python27/PLIST.common
+++ b/lang/python27/PLIST.common
@@ -1835,7 +1835,6 @@
 lib/python${PY_VER_SUFFIX}/lib-dynload/itertools.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/math.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/mmap.so
-${PLIST.nis}lib/python${PY_VER_SUFFIX}/lib-dynload/nis.so
 ${PLIST.no-nis}lib/python${PY_VER_SUFFIX}/lib-dynload/nis_failed.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/operator.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/parser.so
diff --git a/lang/ruby193-base/distinfo b/lang/ruby193-base/distinfo
--- a/lang/ruby193-base/distinfo
+++ b/lang/ruby193-base/distinfo
@@ -27,3 +27,4 @@
 SHA1 (patch-test_rubygems_test__gem.rb) = bb6946eceed2112a7b3d8df84f59c793bdb73bb0
 SHA1 (patch-tool_rbinstall.rb) = f0497085a09b31145f57028b30e335b09acabcff
 SHA1 (patch-vsnprintf.c) = 7bb4ca07381536717518143ef7803634168ca9fa
+SHA1 (patch-aq) = 254f92d175ad31c3e99acb0e0ad4b6cce44e4540
diff --git a/lang/ruby193-base/patches/patch-aq b/lang/ruby193-base/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/lang/ruby193-base/patches/patch-aq
@@ -0,0 +1,29 @@
+diff -r fdf359c0709a configure
+--- configure	Mon Jul 02 14:51:59 2012 +0000
++++ configure	Mon Jul 02 14:53:13 2012 +0000
+@@ -16017,10 +16017,10 @@
+   '^?ELF'*) :
+     rb_cv_binary_elf=yes ;; #(
+   *) :
+-    rb_cv_binary_elf=no ;;
+-esac
+-else
+-  rb_cv_binary_elf=no
++    rb_cv_binary_elf=yes ;;
++esac
++else
++  rb_cv_binary_elf=yes
+ fi
+ rm -f core conftest.err conftest.$ac_objext \
+     conftest$ac_exeext conftest.$ac_ext
+diff -r fdf359c0709a io.c
+--- io.c	Mon Jul 02 14:51:59 2012 +0000
++++ io.c	Mon Jul 02 14:53:13 2012 +0000
+@@ -8123,6 +8123,7 @@
+     VALUE arg[8];
+ #endif
+ #if SIZEOF_VOIDP == 8 && defined(HAVE___SYSCALL) && SIZEOF_INT != 8 /* mainly *BSD */
++#undef __syscall
+ # define SYSCALL __syscall
+ # define NUM2SYSCALLID(x) NUM2LONG(x)
+ # define RETVAL2NUM(x) LONG2NUM(x)
diff --git a/meta-pkgs/xfce4/Makefile b/meta-pkgs/xfce4/Makefile
--- a/meta-pkgs/xfce4/Makefile
+++ b/meta-pkgs/xfce4/Makefile
@@ -16,7 +16,8 @@
 .include "../../print/xfce4-print/buildlink3.mk"
 .include "../../sysutils/xfce4-appfinder/buildlink3.mk"
 .include "../../sysutils/xfce4-thunar/buildlink3.mk"
-.include "../../sysutils/xfce4-volman/buildlink3.mk"
+# [Gregor] volman depends on hal
+#.include "../../sysutils/xfce4-volman/buildlink3.mk"
 .include "../../sysutils/xfce4-xarchiver/buildlink3.mk"
 .include "../../time/xfce4-orage/buildlink3.mk"
 .include "../../wm/xfce4-wm-themes/buildlink3.mk"
diff --git a/misc/pty-redir/distinfo b/misc/pty-redir/distinfo
--- a/misc/pty-redir/distinfo
+++ b/misc/pty-redir/distinfo
@@ -4,3 +4,4 @@
 RMD160 (pty-redir-0.1.tar.gz) = 75347881746dd1419e95e2ada30b23b2d9c0d223
 Size (pty-redir-0.1.tar.gz) = 8405 bytes
 SHA1 (patch-aa) = 28f8a1c22a9bf602e0e447ca45736e7cde6fddd5
+SHA1 (patch-aq) = 17bdbfdef55312079c5ab32069556b50658dd771
diff --git a/misc/pty-redir/patches/patch-aq b/misc/pty-redir/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/misc/pty-redir/patches/patch-aq
@@ -0,0 +1,51 @@
+--- pty-redir.c
++++ pty-redir.c
+@@ -43,30 +43,18 @@
+ #include <fcntl.h>
+ 
+ 
+-#define PTY00 "/dev/ptyXX"
+-#define PTY10 "pqrs"
+-#define PTY01 "0123456789abcdef"
++#define PTY00 "/dev/ptmx"
+ 
+ static int
+ getPtyMaster(char *tty10, char *tty01)
+ {
+-    char *p10;
+-    char *p01;
+     static char dev[] = PTY00;
+     int fd;
+ 
+-    for (p10 = PTY10; *p10 != '\0'; p10++) {
+-        dev[8] = *p10;
+-        for (p01 = PTY01; *p01 != '\0'; p01++) {
+-            dev[9] = *p01;
+             fd = open(dev, O_RDWR);
+             if (fd >= 0) {
+-                *tty10 = *p10;
+-                *tty01 = *p01;
+                 return fd;
+             }
+-        }
+-    }
+     fprintf(stderr,"Ran out of pty.\n");
+     exit(1);
+     return fd;
+@@ -81,8 +69,6 @@
+ 	int i;
+ 	
+ 	fd=getPtyMaster(&a,&b);
+-	fprintf(stdout,"/dev/tty%c%c",a,b);
+-	fflush(stdout);
+ 	if(NULL==(args=malloc((argc)*sizeof(char *))))
+ 	{
+ 		fprintf(stderr,"Virtual memory exhausted\n");
+@@ -95,7 +81,6 @@
+ 	args[argc-1]=NULL; /*End of params*/
+ 	dup2(fd,0);
+ 	dup2(fd,1);
+-	fflush(NULL);
+ 	if(fork())
+ 	{
+ 		exit(0);
diff --git a/net/libfetch/files/ftp.c b/net/libfetch/files/ftp.c
--- a/net/libfetch/files/ftp.c
+++ b/net/libfetch/files/ftp.c
@@ -59,8 +59,10 @@
 
 #ifdef __linux__
 /* Keep this down to Linux, it can create surprises else where. */
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
+#endif
 
 #if HAVE_CONFIG_H
 #include "config.h"
diff --git a/net/libfetch/files/http.c b/net/libfetch/files/http.c
--- a/net/libfetch/files/http.c
+++ b/net/libfetch/files/http.c
@@ -65,8 +65,10 @@
 
 #if defined(__linux__) || defined(__MINT__)
 /* Keep this down to Linux or MiNT, it can create surprises elsewhere. */
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
+#endif
 
 /* Needed for gmtime_r on Interix */
 #define _REENTRANT
diff --git a/net/tightvnc/Makefile b/net/tightvnc/Makefile
--- a/net/tightvnc/Makefile
+++ b/net/tightvnc/Makefile
@@ -28,6 +28,7 @@
 .if !empty(MACHINE_PLATFORM:MNetBSD-*-x86_64)
 _WRAP_EXTRA_ARGS.CC+=	-DDDXTIME -D_XSERVER64 -DGCCUSESGAS
 .endif
+_WRAP_EXTRA_ARGS.CC+=	-DNOFILES_MAX=256
 
 .if ${X11_TYPE} == "modular"
 DEPENDS+=		rgb>=1.0.1:../../x11/rgb
@@ -56,6 +57,11 @@
 
 INSTALLATION_DIRS=	bin ${PKGMANDIR}/man1 share/examples/tightvnc
 
+pre-build:
+	echo '#define DEFAULT_CPP "cpp"' >> ${WRKSRC}/Xvnc/config/imake/imakemdep.h
+	sed -i 's/alloca/malloc/g ; s/-x c -/-x c - -x none/' ${WRKSRC}/Xvnc/config/imake/imake.c
+	sed -i 's/HasFfsl[ \t]*YES/HasFfsl NO/' ${WRKSRC}/Xvnc/config/cf/linux.cf
+
 do-install:
 	${INSTALL_PROGRAM} ${WRKSRC}/Xvnc/programs/Xserver/Xvnc \
 	    ${DESTDIR}${PREFIX}/bin/Xvnc
diff --git a/net/tnftp/files/libedit/tty.c b/net/tnftp/files/libedit/tty.c
--- a/net/tnftp/files/libedit/tty.c
+++ b/net/tnftp/files/libedit/tty.c
@@ -40,6 +40,7 @@
  * tty.c: tty interface stuff
  */
 #include <assert.h>
+#include <strings.h>
 #include "tty.h"
 #include "el.h"
 
diff --git a/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk b/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk
--- a/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk
+++ b/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk
@@ -13,6 +13,7 @@
 .endif
 
 .PHONY:		catinstall maninstall catpages manpages catlinks manlinks cleanman html installhtml cleanhtml
+MKMAN?=no
 .if ${MKMAN} != "no"
 realinstall:	${MANINSTALL}
 .endif
@@ -40,11 +41,10 @@
 TMACDIR?=	${DESTDIR}/usr/share/groff/tmac
 .endif
 HTMLDIR?=	${DESTDIR}/usr/share/man
-CATDEPS?=	${TMACDIR}/mandoc.tmac \
-		${TMACDIR}/doc.tmac
+CATDEPS?=
 MANTARGET?=	cat
-NROFF?=		nroff -Tascii
-GROFF?=		groff -Tascii
+NROFF?=		cat
+GROFF?=		cat
 TBL?=		tbl
 
 
diff --git a/pkgtools/libnbcompat/files/lchflags.c b/pkgtools/libnbcompat/files/lchflags.c
--- a/pkgtools/libnbcompat/files/lchflags.c
+++ b/pkgtools/libnbcompat/files/lchflags.c
@@ -39,6 +39,7 @@
 int
 lchflags(const char *path, unsigned long flags)
 {
+#if 0
 	struct stat psb;
 
 	if (lstat(path, &psb) == -1)
@@ -47,4 +48,6 @@
 		return 0;
 	}
 	return (chflags(path, flags));
+#endif
+	return -1;
 }
diff --git a/pkgtools/libnbcompat/files/unvis.c b/pkgtools/libnbcompat/files/unvis.c
--- a/pkgtools/libnbcompat/files/unvis.c
+++ b/pkgtools/libnbcompat/files/unvis.c
@@ -83,6 +83,8 @@
 #define	isoctal(c)	(((unsigned char)(c)) >= '0' && ((unsigned char)(c)) <= '7')
 #define xtod(c)		(isdigit(c) ? (c - '0') : ((tolower(c) - 'a') + 10))
 
+int __nbcompat_unvis13(char *cp, int c, int *astate, int flag);
+
 int
 unvis(cp, c, astate, flag)
 	char *cp;
diff --git a/pkgtools/pkg_install-info/files/install-info.c b/pkgtools/pkg_install-info/files/install-info.c
--- a/pkgtools/pkg_install-info/files/install-info.c
+++ b/pkgtools/pkg_install-info/files/install-info.c
@@ -21,6 +21,8 @@
 #include "system.h"
 #include <getopt.h>
 
+#include <stdlib.h>
+
 static char *progname = "install-info";
 
 int menu_line_equal (), menu_line_lessp ();
diff --git a/pkgtools/pkg_install-info/files/system.h b/pkgtools/pkg_install-info/files/system.h
--- a/pkgtools/pkg_install-info/files/system.h
+++ b/pkgtools/pkg_install-info/files/system.h
@@ -21,7 +21,9 @@
 #ifndef TEXINFO_SYSTEM_H
 #define TEXINFO_SYSTEM_H
 
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
 
 #include <config.h>
 
diff --git a/pkgtools/pkg_install-info/files/xexit.c b/pkgtools/pkg_install-info/files/xexit.c
--- a/pkgtools/pkg_install-info/files/xexit.c
+++ b/pkgtools/pkg_install-info/files/xexit.c
@@ -19,6 +19,8 @@
 
 #include "system.h"
 
+#include <stdlib.h>
+
 /* SunOS 4.1.1 gets STDC_HEADERS defined, but it doesn't provide
    EXIT_FAILURE.  So far no system has defined one of EXIT_FAILURE and
    EXIT_SUCCESS without the other.  */
diff --git a/pkgtools/pkg_install/files/add/Makefile.in b/pkgtools/pkg_install/files/add/Makefile.in
--- a/pkgtools/pkg_install/files/add/Makefile.in
+++ b/pkgtools/pkg_install/files/add/Makefile.in
@@ -22,7 +22,7 @@
 
 LIBS=		-linstall -lfetch
 .if !empty(SSL_SUPPORT)
-LIBS+=		-lcrypto
+LIBS+=		-lcrypto -lz
 .endif
 LIBS+=		@LIBS@
 
diff --git a/pkgtools/pkg_install/files/admin/Makefile.in b/pkgtools/pkg_install/files/admin/Makefile.in
--- a/pkgtools/pkg_install/files/admin/Makefile.in
+++ b/pkgtools/pkg_install/files/admin/Makefile.in
@@ -30,7 +30,7 @@
 .if empty(BOOTSTRAP)
 LIBS=		-linstall -lfetch
 .if !empty(SSL_SUPPORT)
-LIBS+=		-lcrypto
+LIBS+=		-lcrypto -lz
 CFLAGS+=	-DHAVE_SSL
 .endif
 LIBS+=		@LIBS@
diff --git a/pkgtools/pkg_install/files/info/Makefile.in b/pkgtools/pkg_install/files/info/Makefile.in
--- a/pkgtools/pkg_install/files/info/Makefile.in
+++ b/pkgtools/pkg_install/files/info/Makefile.in
@@ -19,7 +19,7 @@
 .if empty(BOOTSTRAP)
 LIBS=		-linstall -lfetch @LIBS@
 .if !empty(SSL_SUPPORT)
-LIBS+=		-lcrypto
+LIBS+=		-lcrypto -lz
 .endif
 CPPFLAGS=	@CPPFLAGS@ -I. -I$(srcdir) -I../lib
 .else
diff --git a/pkgtools/pkg_install/files/lib/dewey.c b/pkgtools/pkg_install/files/lib/dewey.c
--- a/pkgtools/pkg_install/files/lib/dewey.c
+++ b/pkgtools/pkg_install/files/lib/dewey.c
@@ -39,6 +39,7 @@
 #if HAVE_STDLIB_H
 #include <stdlib.h>
 #endif
+#include <unistd.h> /* for ssize_t */
 
 #include "defs.h"
 #include "dewey.h"
diff --git a/print/a2ps/distinfo b/print/a2ps/distinfo
--- a/print/a2ps/distinfo
+++ b/print/a2ps/distinfo
@@ -7,3 +7,4 @@
 SHA1 (patch-ab) = 7b1f1e3ed2af47e7d9864ec2dbcd7d105f93632a
 SHA1 (patch-ac) = 8e09c4c3b320b58bf12c4266d4d22977b5f9b826
 SHA1 (patch-lib_path-concat.c) = b345a7b16559315223dcbb182e92feccca7e6e2f
+SHA1 (patch-qq) = c6fc0a95e7003c055da2dc1ebea316638a1cc3e9
diff --git a/print/a2ps/patches/patch-qq b/print/a2ps/patches/patch-qq
new file mode 100644
--- /dev/null
+++ b/print/a2ps/patches/patch-qq
@@ -0,0 +1,23 @@
+diff -r b585ce465a54 lib/path-concat.c
+--- lib/path-concat.c	Sun May 06 04:40:07 2012 +0000
++++ lib/path-concat.c	Sun May 06 04:42:23 2012 +0000
+@@ -21,16 +21,15 @@
+ # include <config.h>
+ #endif
+ 
+-#if !defined(HAVE_MEMPCPY) && !defined(__DragonFly__)
+-# define mempcpy(D, S, N) ((void *) ((char *) memcpy (D, S, N) + (N)))
+-#endif
+-
+ #include <stdio.h>
+ #if HAVE_STRING_H
+ # include <string.h>
+ #endif
+ #include <sys/types.h>
+ 
++#if !defined(HAVE_MEMPCPY) && !defined(__DragonFly__)
++# define mempcpy(D, S, N) ((void *) ((char *) memcpy (D, S, N) + (N)))
++#endif
+ /* char *malloc (); */
+ 
+ #ifndef DIRECTORY_SEPARATOR
diff --git a/security/libgpg-error/PLIST b/security/libgpg-error/PLIST
--- a/security/libgpg-error/PLIST
+++ b/security/libgpg-error/PLIST
@@ -8,13 +8,3 @@
 share/common-lisp/source/gpg-error/gpg-error-package.lisp
 share/common-lisp/source/gpg-error/gpg-error.asd
 share/common-lisp/source/gpg-error/gpg-error.lisp
-share/locale/cs/LC_MESSAGES/libgpg-error.mo
-share/locale/de/LC_MESSAGES/libgpg-error.mo
-share/locale/fr/LC_MESSAGES/libgpg-error.mo
-share/locale/it/LC_MESSAGES/libgpg-error.mo
-share/locale/nl/LC_MESSAGES/libgpg-error.mo
-share/locale/pl/LC_MESSAGES/libgpg-error.mo
-share/locale/ro/LC_MESSAGES/libgpg-error.mo
-share/locale/sv/LC_MESSAGES/libgpg-error.mo
-share/locale/vi/LC_MESSAGES/libgpg-error.mo
-share/locale/zh_CN/LC_MESSAGES/libgpg-error.mo
diff --git a/security/openpam/distinfo b/security/openpam/distinfo
--- a/security/openpam/distinfo
+++ b/security/openpam/distinfo
@@ -12,3 +12,4 @@
 SHA1 (patch-ah) = 5cef165a6986e0146f75cc4aa4fe5c0adc2d5042
 SHA1 (patch-ai) = ebd22192a6b34161feac281ade41340493142e2b
 SHA1 (patch-aj) = f106a68e24fabae7353ea4480c75ba84097ec606
+SHA1 (patch-aq) = b8310fd7a88d4a6703ac5b724e5ca1406337b5d1
diff --git a/security/openpam/patches/patch-aq b/security/openpam/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/security/openpam/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r f26e7457d1df lib/openpam_impl.h
+--- lib/openpam_impl.h	Sun May 06 03:31:16 2012 +0000
++++ lib/openpam_impl.h	Sun May 06 03:33:03 2012 +0000
+@@ -42,6 +42,8 @@
+ # include <config.h>
+ #endif
+ 
++#include <sys/types.h>
++
+ #include <security/openpam.h>
+ 
+ extern const char *_pam_func_name[PAM_NUM_PRIMITIVES];
diff --git a/security/openssh/Makefile b/security/openssh/Makefile
--- a/security/openssh/Makefile
+++ b/security/openssh/Makefile
@@ -200,4 +200,7 @@
 	  ${DESTDIR}${EGDIR}/sshd.pam
 .endif
 
+# [Gregor] misdetection 
+CONFIGURE_ARGS+=--without-stackprotect
+
 .include "../../mk/bsd.pkg.mk"
diff --git a/security/openssh/distinfo b/security/openssh/distinfo
--- a/security/openssh/distinfo
+++ b/security/openssh/distinfo
@@ -29,3 +29,4 @@
 SHA1 (patch-av) = dd34ac767d08b989775add25c5c015a2f19fce5c
 SHA1 (patch-aw) = 22d873bfe56464ce8a978082ebce73d441f81e5d
 SHA1 (patch-platform.c) = fcbea87b04d07ef45af5380f9baa024fada9974f
+SHA1 (patch-qq) = ae5da142fcea297d237d5bf32abbed332b3550aa
diff --git a/security/openssh/patches/patch-qq b/security/openssh/patches/patch-qq
new file mode 100644
--- /dev/null
+++ b/security/openssh/patches/patch-qq
@@ -0,0 +1,86 @@
+diff -r 989278cadbd7 loginrec.c
+--- loginrec.c	Sat Jul 14 01:30:04 2012 +0000
++++ loginrec.c	Sat Jul 14 01:35:20 2012 +0000
+@@ -777,6 +777,7 @@
+ 	strncpy(utx->ut_host, li->hostname,
+ 	    MIN_SIZEOF(utx->ut_host, li->hostname));
+ # endif
++#if 0
+ # ifdef HAVE_ADDR_IN_UTMPX
+ 	/* this is just a 32-bit IP address */
+ 	if (li->hostaddr.sa.sa_family == AF_INET)
+@@ -795,6 +796,7 @@
+ 		}
+ 	}
+ # endif
++#endif
+ # ifdef HAVE_SYSLEN_IN_UTMPX
+ 	/* ut_syslen is the length of the utx_host string */
+ 	utx->ut_syslen = MIN(strlen(li->hostname), sizeof(utx->ut_host));
+@@ -1703,8 +1705,10 @@
+ 		ipv64_normalise_mapped(&from, &fromlen);
+ 		if (from.ss_family == AF_INET) {
+ 			a4 = (struct sockaddr_in *)&from;
++#if 0
+ 			memcpy(&ut.ut_addr, &(a4->sin_addr),
+ 			    MIN_SIZEOF(ut.ut_addr, a4->sin_addr));
++#endif
+ 		}
+ #ifdef HAVE_ADDR_V6_IN_UTMP
+ 		if (from.ss_family == AF_INET6) {
+diff -r 989278cadbd7 misc.c
+--- misc.c	Sat Jul 14 01:30:04 2012 +0000
++++ misc.c	Sat Jul 14 01:35:20 2012 +0000
+@@ -54,6 +54,8 @@
+ #include <net/if.h>
+ #endif
+ 
++#include <sys/time.h>
++
+ #include "xmalloc.h"
+ #include "misc.h"
+ #include "log.h"
+diff -r 989278cadbd7 sftp-server.c
+--- sftp-server.c	Sat Jul 14 01:30:04 2012 +0000
++++ sftp-server.c	Sat Jul 14 01:35:20 2012 +0000
+@@ -51,6 +51,10 @@
+ #include "sftp.h"
+ #include "sftp-common.h"
+ 
++#ifndef NFDBITS
++#define NFDBITS (8*sizeof(unsigned long))
++#endif
++
+ /* helper */
+ #define get_int64()			buffer_get_int64(&iqueue);
+ #define get_int()			buffer_get_int(&iqueue);
+diff -r 989278cadbd7 ssh-pkcs11-helper.c
+--- ssh-pkcs11-helper.c	Sat Jul 14 01:30:04 2012 +0000
++++ ssh-pkcs11-helper.c	Sat Jul 14 01:35:20 2012 +0000
+@@ -37,6 +37,11 @@
+ #include "authfd.h"
+ #include "ssh-pkcs11.h"
+ 
++#include <sys/param.h>
++#ifndef NFDBITS
++#define NFDBITS (8*sizeof(unsigned long))
++#endif
++
+ #ifdef ENABLE_PKCS11
+ 
+ /* borrows code from sftp-server and ssh-agent */
+diff -r 989278cadbd7 ssh.h
+--- ssh.h	Sat Jul 14 01:30:04 2012 +0000
++++ ssh.h	Sat Jul 14 01:35:20 2012 +0000
+@@ -11,6 +11,11 @@
+  * incompatible with the protocol description in the RFC file, it must be
+  * called by a name other than "ssh" or "Secure Shell".
+  */
++ 
++#include <sys/param.h>
++#ifndef NFDBITS
++#define NFDBITS (8*sizeof(unsigned long))
++#endif
+ 
+ /* Cipher used for encrypting authentication files. */
+ #define SSH_AUTHFILE_CIPHER	SSH_CIPHER_3DES
diff --git a/security/openssl/Makefile b/security/openssl/Makefile
--- a/security/openssl/Makefile
+++ b/security/openssl/Makefile
@@ -61,6 +61,7 @@
 # back to ./config's autodetection if not.
 .  if defined(OPENSSL_MACHINE_ARCH) && !empty(OPENSSL_MACHINE_ARCH)
 CONFIGURE_SCRIPT=	./Configure
+CONFIGURE_ARGS+=	no-shared no-dso zlib enable-md2 -Wa,--noexecstack
 .    if !empty(CC_VERSION:Mgcc*)
 CONFIGURE_ARGS+=	solaris${${ABI}==64:?64:}-${OPENSSL_MACHINE_ARCH}-gcc
 .    else
@@ -154,8 +155,16 @@
 
 INSTALLATION_DIRS+=	share/examples/openssl
 
+pre-build:
+	touch ${WRKSRC}/include/memory.h
+	cd ${WRKSRC} && $(MAKE) depend
+
 # Fix the path to perl in various scripts.
 pre-configure:
 	cd ${WRKSRC} && ${PERL5} util/perlpath.pl ${PERL5}
-
+	sed -i '/^"linux-x86_64/s/-DTERMIO/-DTERMIOS/' ${WRKSRC}/Configure
+	sed -i '/^"linux-elf/s/-DTERMIO/-DTERMIOS/' ${WRKSRC}/Configure
+	sed -i 's/defined(linux)/0/' ${WRKSRC}/crypto/ui/ui_openssl.c
+	sed -i 's/\!defined(_UNICOS)/0/' ${WRKSRC}/apps/speed.c
+	
 .include "../../mk/bsd.pkg.mk"
diff --git a/security/p5-Crypt-Rijndael/distinfo b/security/p5-Crypt-Rijndael/distinfo
--- a/security/p5-Crypt-Rijndael/distinfo
+++ b/security/p5-Crypt-Rijndael/distinfo
@@ -3,3 +3,4 @@
 SHA1 (Crypt-Rijndael-1.09.tar.gz) = 2edc7836a62c01c6db668586f907f8f57170a970
 RMD160 (Crypt-Rijndael-1.09.tar.gz) = a0fa349631ce7a5555a3f6a9fe6c23b1a745858d
 Size (Crypt-Rijndael-1.09.tar.gz) = 28994 bytes
+SHA1 (patch-aq) = 4780039094f3df9bde4617f4f26c11a6f23b56bf
diff --git a/security/p5-Crypt-Rijndael/patches/patch-aq b/security/p5-Crypt-Rijndael/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/security/p5-Crypt-Rijndael/patches/patch-aq
@@ -0,0 +1,14 @@
+diff -r d81bee0804d7 rijndael.h
+--- rijndael.h	Mon Jul 02 16:04:39 2012 +0000
++++ rijndael.h	Mon Jul 02 16:05:26 2012 +0000
+@@ -74,8 +74,8 @@
+ /* I expect this to be the usual case */
+ #if ! defined(_CRYPT_RIJNDAEL_H_TYPES) && ( defined(_SYS_TYPES_H) || defined(_SYS_TYPES_H_) )   
+ 	#define _CRYPT_RIJNDAEL_H_TYPES
+-	typedef __uint32_t UINT32;
+-	typedef __uint8_t  UINT8;
++	typedef uint32_t UINT32;
++	typedef uint8_t  UINT8;
+ #endif
+ 
+ #if defined(__CYGWIN__) && ! defined(_CRYPT_RIJNDAEL_H_TYPES)
diff --git a/security/policykit/distinfo b/security/policykit/distinfo
--- a/security/policykit/distinfo
+++ b/security/policykit/distinfo
@@ -19,3 +19,4 @@
 SHA1 (patch-an) = a0a54235fc72a52ace72a0d04df6c1383f7c583c
 SHA1 (patch-ao) = 0f9a87653f4902285c7420f6269e302eca896a1b
 SHA1 (patch-authdb) = 449b8ac50f81cc0234e17e64bc9892ade7119028
+SHA1 (patch-aq) = 07d7453aba399fd3b61e05a46ec20f6080be7fde
diff --git a/security/policykit/patches/patch-aq b/security/policykit/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/security/policykit/patches/patch-aq
@@ -0,0 +1,11 @@
+diff -r 1ceaf4291bd1 src/polkit-dbus/polkit-resolve-exe-helper.c
+--- src/polkit-dbus/polkit-resolve-exe-helper.c	Sun May 06 03:39:34 2012 +0000
++++ src/polkit-dbus/polkit-resolve-exe-helper.c	Sun May 06 03:39:59 2012 +0000
+@@ -52,6 +52,7 @@
+ #include <utime.h>
+ #include <fcntl.h>
+ #include <dirent.h>
++#include <limits.h>
+ 
+ #include <polkit-dbus/polkit-dbus.h>
+ #include <polkit/polkit-private.h>
diff --git a/shells/bash/Makefile b/shells/bash/Makefile
--- a/shells/bash/Makefile
+++ b/shells/bash/Makefile
@@ -78,6 +78,9 @@
 .include "../../devel/readline/buildlink3.mk"
 .endif
 
+# [Gregor] mischeck for musl, safe everywhere
+CONFIGURE_ARGS+=		--without-bash-malloc
+
 post-install:
 	${INSTALL_MAN} ${WRKSRC}/doc/bash.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
 	${INSTALL_MAN} ${WRKSRC}/doc/bashbug.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
diff --git a/shells/dash/Makefile b/shells/dash/Makefile
--- a/shells/dash/Makefile
+++ b/shells/dash/Makefile
@@ -16,4 +16,6 @@
 USE_TOOLS+=		gmake
 PKG_SHELL=		bin/dash
 
+CONFIGURE_ARGS+=	CFLAGS="-O2 -g -DEXECCMD=1 -DCOMMANDCMD=2"
+
 .include "../../mk/bsd.pkg.mk"
diff --git a/sysutils/coreutils/Makefile b/sysutils/coreutils/Makefile
--- a/sysutils/coreutils/Makefile
+++ b/sysutils/coreutils/Makefile
@@ -39,6 +39,8 @@
 #MESSAGE_SUBST+=	OPSYS=${OPSYS:Q}
 #.endif
 
+CFLAGS+=		-DSLOW_BUT_NO_HACKS
+
 .include "../../converters/libiconv/buildlink3.mk"
 .include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
diff --git a/sysutils/coreutils/distinfo b/sysutils/coreutils/distinfo
--- a/sysutils/coreutils/distinfo
+++ b/sysutils/coreutils/distinfo
@@ -5,3 +5,4 @@
 Size (coreutils-8.13.tar.gz) = 11660808 bytes
 SHA1 (patch-lib-fflush.c) = e52e0da2127c5debbb14e967b1e206226efdaa4a
 SHA1 (patch-lib-fseeko.c) = ab62b8c7faa9a601191475c040c7466e1bd52fe1
+SHA1 (patch-aq) = 02186df2372b3317cc2df6a8230cc198ac3eb4e9
diff --git a/sysutils/coreutils/patches/patch-aq b/sysutils/coreutils/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/coreutils/patches/patch-aq
@@ -0,0 +1,23 @@
+diff -r fb0a1c120d18 lib/fseterr.c
+--- lib/fseterr.c	Sat May 05 16:00:33 2012 +0000
++++ lib/fseterr.c	Sat May 05 16:05:52 2012 +0000
+@@ -71,6 +71,6 @@
+     }
+   errno = saved_errno;
+ #else
+- #error "Please port gnulib fseterr.c to your platform! Look at the definitions of ferror and clearerr on your system, then report this to bug-gnulib."
++  return 1;
+ #endif
+ }
+diff -r fb0a1c120d18 src/dd.c
+--- src/dd.c	Sat May 05 16:00:33 2012 +0000
++++ src/dd.c	Sat May 05 16:05:52 2012 +0000
+@@ -1375,7 +1375,7 @@
+    The offending behavior has been confirmed with an Exabyte SCSI tape
+    drive accessed via /dev/nst0 on both Linux 2.2.17 and 2.4.16 kernels.  */
+ 
+-#ifdef __linux__
++#if 0
+ 
+ # include <sys/mtio.h>
+ 
diff --git a/sysutils/gamin/distinfo b/sysutils/gamin/distinfo
--- a/sysutils/gamin/distinfo
+++ b/sysutils/gamin/distinfo
@@ -16,3 +16,4 @@
 SHA1 (patch-server_gam__node.h) = 49a0d381b4900564c4f0ba7a654eb22e294c0a16
 SHA1 (patch-server_gam__subscription.c) = 58f6c10e4e1e899d097662536ea31bf8a692b114
 SHA1 (patch-server_gam__subscription.h) = 6347865210aa0338bc8b0a81fc671824c786fd26
+SHA1 (patch-aq) = 6d30c2cb4ef9504a704a50168899a11c8c0db78a
diff --git a/sysutils/gamin/patches/patch-aq b/sysutils/gamin/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/gamin/patches/patch-aq
@@ -0,0 +1,15 @@
+diff -r a17041040b0f server/gam_dnotify.c
+--- server/gam_dnotify.c	Sun May 06 05:06:48 2012 +0000
++++ server/gam_dnotify.c	Sun May 06 05:13:29 2012 +0000
+@@ -19,6 +19,11 @@
+ 
+ #include "server_config.h"
+ #define _GNU_SOURCE
++#if defined(__linux__) && !defined(__GLIBC__)
++/* need to do this nasty hack until/unless musl exposes the right #defines */
++#define HAVE_ARCH_STRUCT_FLOCK
++#include <linux/fcntl.h>
++#endif
+ #include <fcntl.h>
+ #include <signal.h>
+ #include <unistd.h>
diff --git a/sysutils/libpciaccess/distinfo b/sysutils/libpciaccess/distinfo
--- a/sysutils/libpciaccess/distinfo
+++ b/sysutils/libpciaccess/distinfo
@@ -5,3 +5,4 @@
 Size (libpciaccess-0.11.0.tar.bz2) = 295639 bytes
 SHA1 (patch-aa) = 6bb844c0696c8a86c6033605339e82da34164bdc
 SHA1 (patch-ac) = c2df07d306203f3236e3a79caf4eedc5104d6f78
+SHA1 (patch-aq) = 0f0850644192c53cab4e5a8105e6920ff46e8171
diff --git a/sysutils/libpciaccess/patches/patch-aq b/sysutils/libpciaccess/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/libpciaccess/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 6157f7aaf390 src/linux_sysfs.c
+--- src/linux_sysfs.c	Thu Jul 19 02:37:38 2012 +0000
++++ src/linux_sysfs.c	Thu Jul 19 02:38:32 2012 +0000
+@@ -44,6 +44,8 @@
+ #include <dirent.h>
+ #include <errno.h>
+ 
++#include <limits.h>
++
+ #include "config.h"
+ 
+ #ifdef HAVE_MTRR
diff --git a/sysutils/pciutils/distinfo b/sysutils/pciutils/distinfo
--- a/sysutils/pciutils/distinfo
+++ b/sysutils/pciutils/distinfo
@@ -7,3 +7,4 @@
 SHA1 (patch-ab) = 1448bb418cdc908eb6652cc98994d23d0d78cea0
 SHA1 (patch-ac) = c917e8ad2d1e87fa65a2d8f3256afd40d5a87396
 SHA1 (patch-ad) = bfe9069e96fd46798afec73c93ad71db3a0e0e37
+SHA1 (patch-aq) = b29a7e176a23456b6c509e5c40326d5a59050b8c
diff --git a/sysutils/pciutils/patches/patch-aq b/sysutils/pciutils/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/pciutils/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 63e97996f76b lib/configure
+--- lib/configure	Sun May 06 02:42:29 2012 +0000
++++ lib/configure	Sun May 06 02:42:58 2012 +0000
+@@ -68,7 +68,7 @@
+ 		echo >>$c '#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"'
+ 		echo >>$c '#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"'
+ 		case $cpu in
+-				i?86|x86_64)	echo_n " i386-ports"
++				disabled-i?86|disabled-x86_64)	echo_n " i386-ports"
+ 						echo >>$c '#define PCI_HAVE_PM_INTEL_CONF'
+ 						;;
+ 		esac
diff --git a/sysutils/xfce4-thunar/distinfo b/sysutils/xfce4-thunar/distinfo
--- a/sysutils/xfce4-thunar/distinfo
+++ b/sysutils/xfce4-thunar/distinfo
@@ -7,3 +7,4 @@
 SHA1 (patch-ab) = e5c3f1e853bf59f69047c7160a3e29df54298879
 SHA1 (patch-ac) = 745f240ff44b0b21eb727b0a9887c9d8f1ee8a47
 SHA1 (patch-thunar-vfs_thunar-vfs-io-jobs.c) = 8529460dea3353b7b8bb034c65c95e4561f61aab
+SHA1 (patch-aq) = dd802620cec6c8d76a4ae09e5a155f2b01fd3558
diff --git a/sysutils/xfce4-thunar/patches/patch-aq b/sysutils/xfce4-thunar/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/xfce4-thunar/patches/patch-aq
@@ -0,0 +1,11 @@
+diff -r f88137634fb9 tdb/tdb.h
+--- tdb/tdb.h	Sun May 06 09:10:30 2012 +0000
++++ tdb/tdb.h	Sun May 06 11:32:55 2012 +0000
+@@ -29,6 +29,7 @@
+ 
+ #include <signal.h>
+ #include <unistd.h>
++#include <sys/types.h>
+ 
+ #ifdef  __cplusplus
+ extern "C" {
diff --git a/textproc/groff/PLIST b/textproc/groff/PLIST
--- a/textproc/groff/PLIST
+++ b/textproc/groff/PLIST
@@ -15,9 +15,6 @@
 bin/grops
 bin/grotty
 bin/hpftodit
-bin/${GNU_PREFIX}indxbib
-bin/lkbib
-bin/${GNU_PREFIX}lookbib
 bin/mmroff
 bin/${GNU_PREFIX}neqn
 bin/${GNU_PREFIX}nroff
@@ -28,7 +25,6 @@
 bin/post-grohtml
 bin/pre-grohtml
 bin/preconv
-bin/${GNU_PREFIX}refer
 bin/roff2dvi
 bin/roff2html
 bin/roff2pdf
@@ -40,11 +36,11 @@
 bin/tfmtodit
 bin/${GNU_PREFIX}troff
 info/groff.info
-lib/groff/groffer/func.pl
-lib/groff/groffer/man.pl
-lib/groff/groffer/perl_test.pl
-lib/groff/groffer/split_env.sh
-lib/groff/groffer/version.sh
+@comment lib/groff/groffer/func.pl
+@comment lib/groff/groffer/man.pl
+@comment lib/groff/groffer/perl_test.pl
+@comment lib/groff/groffer/split_env.sh
+@comment lib/groff/groffer/version.sh
 man/man1/addftinfo.1
 man/man1/afmtodit.1
 man/man1/chem.1
diff --git a/textproc/hunspell/distinfo b/textproc/hunspell/distinfo
--- a/textproc/hunspell/distinfo
+++ b/textproc/hunspell/distinfo
@@ -6,3 +6,4 @@
 SHA1 (patch-aa) = b03a51d88bd57786f4a0ac01a3b760b4d8ba385a
 SHA1 (patch-ab) = 01ff90fc2ebf1049d791a0669ddcc3f5495729c0
 SHA1 (patch-ac) = 27cad6078b55bce7144f462ddebac9887392c93b
+SHA1 (patch-aq) = ac8d1f4468d017c841d33ab7cb073944921f03bf
diff --git a/textproc/hunspell/patches/patch-aq b/textproc/hunspell/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/textproc/hunspell/patches/patch-aq
@@ -0,0 +1,24 @@
+diff -r 006c3f933ef3 src/tools/munch.c
+--- src/tools/munch.c	Mon May 07 23:49:28 2012 +0000
++++ src/tools/munch.c	Mon May 07 23:50:25 2012 +0000
+@@ -8,7 +8,7 @@
+ #include <sys/types.h>
+ #include <sys/stat.h>
+ #include <fcntl.h>
+-#ifdef __linux__
++#if defined(__linux__) && defined(__GLIBC__)
+ #include <error.h>
+ #include <errno.h>
+ #include <sys/mman.h>
+diff -r 006c3f933ef3 src/tools/unmunch.c
+--- src/tools/unmunch.c	Mon May 07 23:49:28 2012 +0000
++++ src/tools/unmunch.c	Mon May 07 23:50:25 2012 +0000
+@@ -10,7 +10,7 @@
+ #include <sys/types.h>
+ #include <sys/stat.h>
+ #include <fcntl.h>
+-#ifdef __linux__
++#if defined(__linux__) && defined(__GLIBC__)
+ #include <error.h>
+ #include <errno.h>
+ #include <sys/mman.h>
diff --git a/textproc/libxml2/Makefile b/textproc/libxml2/Makefile
--- a/textproc/libxml2/Makefile
+++ b/textproc/libxml2/Makefile
@@ -24,6 +24,7 @@
 CONFIGURE_ARGS+=	--with-iconv=${BUILDLINK_PREFIX.iconv}
 CONFIGURE_ARGS+=	--with-html-subdir=libxml2
 CONFIGURE_ARGS+=	--without-python
+CONFIGURE_ARGS+=	--without-threads
 
 .include "options.mk"
 
@@ -43,6 +44,9 @@
 SUBST_SED.cat=	-e "s,@@SGML_DEFAULT_CATALOG@@,${SGML_DEFAULT_CATALOG},g"
 SUBST_SED.cat+=	-e "s,@@XML_DEFAULT_CATALOG@@,${XML_DEFAULT_CATALOG},g"
 
+pre-build:
+	sed 's/NO_ADDRESS/-1/' -i ${WRKSRC}/nanohttp.c
+
 .include "../../mk/pthread.buildlink3.mk"
 .include "../../devel/zlib/buildlink3.mk"
 .include "../../converters/libiconv/buildlink3.mk"
diff --git a/textproc/xmlrpc-c/Makefile b/textproc/xmlrpc-c/Makefile
--- a/textproc/xmlrpc-c/Makefile
+++ b/textproc/xmlrpc-c/Makefile
@@ -40,5 +40,7 @@
 
 .include "../../www/curl/buildlink3.mk"
 .include "../../www/libwww/buildlink3.mk"
+.include "../../converters/libiconv/buildlink3.mk"
+.include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../mk/pthread.buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
diff --git a/textproc/xmlrpc-c/distinfo b/textproc/xmlrpc-c/distinfo
--- a/textproc/xmlrpc-c/distinfo
+++ b/textproc/xmlrpc-c/distinfo
@@ -13,3 +13,4 @@
 SHA1 (patch-lib_curl__transport_curlmulti.c) = 1cea4c43b1ed3dcf73827afc14d7bb9267371121
 SHA1 (patch-lib_curl__transport_curltransaction.c) = 1dcf46c0dd1dd8e7fdfbb6c81e00b326f6d0b1f9
 SHA1 (patch-lib_curl__transport_xmlrpc__curl__transport.c) = ea132ebb46ff0ce9893d64e9c9205d5a155220ef
+SHA1 (patch-aq) = 434c86fea4f13a8d0ba92e82b11bf3476a3017da
diff --git a/textproc/xmlrpc-c/patches/patch-aq b/textproc/xmlrpc-c/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/textproc/xmlrpc-c/patches/patch-aq
@@ -0,0 +1,48 @@
+diff -r 89442310b346 examples/interrupted_client.c
+--- examples/interrupted_client.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/interrupted_client.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -2,6 +2,8 @@
+    both by timeout and by control-C.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <unistd.h>
+ #include <stdio.h>
+diff -r 89442310b346 examples/interrupted_server.c
+--- examples/interrupted_server.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/interrupted_server.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -7,6 +7,8 @@
+    simpler code, but it is not interruptible with SIGTERM.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <signal.h>
+diff -r 89442310b346 examples/xmlrpc_inetd_server.c
+--- examples/xmlrpc_inetd_server.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/xmlrpc_inetd_server.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -21,6 +21,8 @@
+    respond to the client, then exit.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <signal.h>
+diff -r 89442310b346 examples/xmlrpc_loop_server.c
+--- examples/xmlrpc_loop_server.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/xmlrpc_loop_server.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -6,6 +6,8 @@
+    less control over how the requests are served.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <signal.h>
diff --git a/wm/bsetroot/PLIST b/wm/bsetroot/PLIST
--- a/wm/bsetroot/PLIST
+++ b/wm/bsetroot/PLIST
@@ -3,37 +3,3 @@
 bin/bsetroot
 man/man1/bsetbg.1
 man/man1/bsetroot.1
-share/bsetroot/nls/POSIX
-share/bsetroot/nls/US_ASCII
-share/bsetroot/nls/de
-share/bsetroot/nls/en
-share/bsetroot/nls/en_GB
-share/bsetroot/nls/en_US
-share/bsetroot/nls/es
-share/bsetroot/nls/es_MX
-share/bsetroot/nls/fr
-share/bsetroot/nls/it
-share/bsetroot/nls/nl
-share/bsetroot/nls/C/blackbox.cat
-share/bsetroot/nls/da_DK/blackbox.cat
-share/bsetroot/nls/de_DE/blackbox.cat
-share/bsetroot/nls/es_AR/blackbox.cat
-share/bsetroot/nls/es_ES/blackbox.cat
-share/bsetroot/nls/fr_FR/blackbox.cat
-share/bsetroot/nls/hu_HU/blackbox.cat
-share/bsetroot/nls/it_IT/blackbox.cat
-share/bsetroot/nls/ja_JP/blackbox.cat
-share/bsetroot/nls/ko_KR/blackbox.cat
-share/bsetroot/nls/lv_LV/blackbox.cat
-share/bsetroot/nls/nl_NL/blackbox.cat
-share/bsetroot/nls/no_NO/blackbox.cat
-share/bsetroot/nls/pl_PL/blackbox.cat
-share/bsetroot/nls/pt_BR/blackbox.cat
-share/bsetroot/nls/ro_RO/blackbox.cat
-share/bsetroot/nls/ru_RU/blackbox.cat
-share/bsetroot/nls/sk_SK/blackbox.cat
-share/bsetroot/nls/sl_SI/blackbox.cat
-share/bsetroot/nls/sv_SE/blackbox.cat
-share/bsetroot/nls/uk_UA/blackbox.cat
-share/bsetroot/nls/zh_CN/blackbox.cat
-share/bsetroot/nls/zh_TW/blackbox.cat
diff --git a/www/firefox/Makefile b/www/firefox/Makefile
--- a/www/firefox/Makefile
+++ b/www/firefox/Makefile
@@ -19,6 +19,7 @@
 
 LDFLAGS.DragonFly=	-lplc4 -lnspr4
 LDFLAGS.FreeBSD=	-lplc4 -lnspr4
+LDFLAGS.Linux=	-lplc4 -lnspr4
 
 .include "../../devel/xulrunner/mozilla-common.mk"
 .include "options.mk"
@@ -31,6 +32,9 @@
 MOZILLA_NAME=	Browser
 .endif
 
+# [Gregor] To make some patches work
+CFLAGS+=		-D__musl__
+
 pre-configure:
 	cd ${WRKSRC} && autoconf
 # XXX not built here, just to make check-portability.awk shut up.
diff --git a/www/webkit-gtk/Makefile b/www/webkit-gtk/Makefile
--- a/www/webkit-gtk/Makefile
+++ b/www/webkit-gtk/Makefile
@@ -9,5 +9,8 @@
 
 CONFIGURE_ARGS+=	--with-gtk=2.0
 
+# [Gregor] For patches
+CFLAGS+=	-D__musl__
+
 .include "../../x11/gtk2/buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
diff --git a/x11/libFS/Makefile b/x11/libFS/Makefile
--- a/x11/libFS/Makefile
+++ b/x11/libFS/Makefile
@@ -25,6 +25,8 @@
 CONFIGURE_ARGS+=	--disable-malloc0returnsnull
 .endif
 
+CONFIGURE_ARGS+=	CFLAGS="-O2 -g -DNOFILES_MAX=256"
+
 .include "../../x11/fontsproto/buildlink3.mk"
 .include "../../x11/xproto/buildlink3.mk"
 .include "../../x11/xtrans/buildlink3.mk"
diff --git a/x11/libFS/distinfo b/x11/libFS/distinfo
--- a/x11/libFS/distinfo
+++ b/x11/libFS/distinfo
@@ -3,3 +3,4 @@
 SHA1 (libFS-1.0.3.tar.bz2) = 2b33fa17369605303ae70a7de1aa681879249c92
 RMD160 (libFS-1.0.3.tar.bz2) = a4c1811de0cba5fd40a083031c44a951c240dd3b
 Size (libFS-1.0.3.tar.bz2) = 284331 bytes
+SHA1 (patch-aq) = 75f64d23fe0ffdc5dca48fcba982539d34caf42b
diff --git a/x11/libFS/patches/patch-aq b/x11/libFS/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/libFS/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 5f55f742a811 src/FSConnServ.c
+--- src/FSConnServ.c	Mon May 07 19:40:35 2012 +0000
++++ src/FSConnServ.c	Mon May 07 19:44:53 2012 +0000
+@@ -49,6 +49,8 @@
+ 
+ */
+ 
++#define _GNU_SOURCE
++
+ #ifdef HAVE_CONFIG_H
+ #include <config.h>
+ #endif
diff --git a/x11/libXfont/Makefile b/x11/libXfont/Makefile
--- a/x11/libXfont/Makefile
+++ b/x11/libXfont/Makefile
@@ -20,6 +20,7 @@
 PKGCONFIG_OVERRIDE+=	xfont.pc.in
 
 CONFIGURE_ARGS+=	--disable-devel-docs
+CONFIGURE_ARGS+=	CFLAGS="-O2 -g -DNOFILES_MAX=256"
 
 .include "../../devel/zlib/buildlink3.mk"
 .include "../../graphics/freetype2/buildlink3.mk"
diff --git a/x11/libdrm/distinfo b/x11/libdrm/distinfo
--- a/x11/libdrm/distinfo
+++ b/x11/libdrm/distinfo
@@ -7,3 +7,4 @@
 SHA1 (patch-ac) = 67c998df7dfc0dabc86320ea6d015cede3e464ea
 SHA1 (patch-ad) = ea3ef8de8db782ddab23f182f1e6bd9d8f612a78
 SHA1 (patch-libkms_linux.c) = 737d0a8e7b742ee9d24fec94c72de6b4915dd675
+SHA1 (patch-aq) = 87e114775f5761986528f8c0c87ebaefe52d593b
diff --git a/x11/libdrm/patches/patch-aq b/x11/libdrm/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/libdrm/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 3f68fbb9449a intel/intel_bufmgr_fake.c
+--- intel/intel_bufmgr_fake.c	Mon May 07 20:01:11 2012 +0000
++++ intel/intel_bufmgr_fake.c	Mon May 07 20:01:34 2012 +0000
+@@ -51,6 +51,8 @@
+ #include "mm.h"
+ #include "libdrm_lists.h"
+ 
++#include <strings.h>
++
+ /* Support gcc's __FUNCTION__ for people using other compilers */
+ #if !defined(__GNUC__) && !defined(__FUNCTION__)
+ # define __FUNCTION__ __func__ /* C99 */
diff --git a/x11/libxcb/Makefile b/x11/libxcb/Makefile
--- a/x11/libxcb/Makefile
+++ b/x11/libxcb/Makefile
@@ -74,6 +74,8 @@
 
 PYTHON_VERSIONS_ACCEPTED= 27 26 25
 
+BUILD_DEPENDS+=		py27-cElementTree-[0-9]*:../../textproc/py-cElementTree
+
 post-extract:
 	${CHMOD} -R go-w ${WRKSRC}
 
diff --git a/x11/libxfce4menu/distinfo b/x11/libxfce4menu/distinfo
--- a/x11/libxfce4menu/distinfo
+++ b/x11/libxfce4menu/distinfo
@@ -3,3 +3,4 @@
 SHA1 (libxfce4menu-4.6.1.tar.bz2) = 17348f661c04e8d8d633b8831638d38987269e9a
 RMD160 (libxfce4menu-4.6.1.tar.bz2) = 978071d28288bd737219e4a9c4185937b2fbf50b
 Size (libxfce4menu-4.6.1.tar.bz2) = 404201 bytes
+SHA1 (patch-aq) = 819f9876969820c0b8beb98a0cd7580eb59237dc
diff --git a/x11/libxfce4menu/patches/patch-aq b/x11/libxfce4menu/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/libxfce4menu/patches/patch-aq
@@ -0,0 +1,11 @@
+diff -r 24e4986df43b tdb/tdb.h
+--- tdb/tdb.h	Sun May 06 11:48:51 2012 +0000
++++ tdb/tdb.h	Sun May 06 11:49:25 2012 +0000
+@@ -29,6 +29,7 @@
+ 
+ #include <signal.h>
+ #include <unistd.h>
++#include <sys/types.h>
+ 
+ #ifdef  __cplusplus
+ extern "C" {
diff --git a/x11/pixman/Makefile b/x11/pixman/Makefile
--- a/x11/pixman/Makefile
+++ b/x11/pixman/Makefile
@@ -42,5 +42,8 @@
 post-extract:
 	${CP} ${FILESDIR}/pixman.sym ${WRKDIR}
 	${CP} ${FILESDIR}/solaris-hwcap.mapfile ${WRKSRC}/pixman
-
+	
+pre-configure:
+	sed -i 's/support_for__thread=yes/support_for__thread=no/' ${WRKSRC}/configure	
+	
 .include "../../mk/bsd.pkg.mk"
diff --git a/x11/vte/distinfo b/x11/vte/distinfo
--- a/x11/vte/distinfo
+++ b/x11/vte/distinfo
@@ -9,3 +9,4 @@
 SHA1 (patch-af) = 69ea7e022f6c495c3c9af681d05644b9d70e7bdc
 SHA1 (patch-ah) = 28cee2661439d7d6f1959ebf3c7d1b7b3fe0a764
 SHA1 (patch-ai) = 986dd7c2a5778c9c35100901cfaa25acd0cbdc32
+SHA1 (patch-aq) = 8262b1ccde6869c18a27994dd19e469b5430b05a
diff --git a/x11/vte/patches/patch-aq b/x11/vte/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/vte/patches/patch-aq
@@ -0,0 +1,19 @@
+diff -r fecfb00f45d9 src/pty.c
+--- src/pty.c	Sun May 06 12:32:48 2012 +0000
++++ src/pty.c	Sun May 06 12:34:08 2012 +0000
+@@ -290,6 +290,7 @@
+ #endif
+ 
+ #ifdef HAVE_STROPTS_H
++#if 0
+ 	if (isastream (fd) == 1) {
+ 		if ((ioctl(fd, I_FIND, "ptem") == 0) &&
+ 				(ioctl(fd, I_PUSH, "ptem") == -1)) {
+@@ -306,6 +307,7 @@
+ 		}
+ 	}
+ #endif
++#endif
+ 
+ 	/* now setup child I/O through the tty */
+ 	if (fd != STDIN_FILENO) {
diff --git a/x11/xinit/distinfo b/x11/xinit/distinfo
--- a/x11/xinit/distinfo
+++ b/x11/xinit/distinfo
@@ -3,3 +3,4 @@
 SHA1 (xinit-1.2.1.tar.bz2) = baf36374aa1632054816a8281e55e1539459664b
 RMD160 (xinit-1.2.1.tar.bz2) = 993b068e4758521ed930bd78069c291b68d88037
 Size (xinit-1.2.1.tar.bz2) = 141441 bytes
+SHA1 (patch-aq) = 98a0f800293163faa1392090842a6994485e8204
diff --git a/x11/xinit/patches/patch-aq b/x11/xinit/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/xinit/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 55926f90284b xinit.c
+--- xinit.c	Sun May 06 12:44:26 2012 +0000
++++ xinit.c	Sun May 06 12:44:44 2012 +0000
+@@ -153,7 +153,7 @@
+ static char *program = NULL;
+ static Display *xd = NULL;			/* server connection */
+ #ifndef SYSV
+-#if defined(__CYGWIN__) || defined(SVR4) || defined(_POSIX_SOURCE) || defined(CSRG_BASED) || defined(__UNIXOS2__) || defined(Lynx) || defined(__APPLE__)
++#if defined(__CYGWIN__) || defined(SVR4) || defined(_POSIX_SOURCE) || defined(CSRG_BASED) || defined(__UNIXOS2__) || defined(Lynx) || defined(__APPLE__) || 1
+ int status;
+ #else
+ union wait	status;
