# HG changeset patch
# Parent 5c00f38e7133a9f1dbbc25848fde5fce7a27a7dc

diff --git a/audio/alsa-lib/distinfo b/audio/alsa-lib/distinfo
--- a/audio/alsa-lib/distinfo
+++ b/audio/alsa-lib/distinfo
@@ -42,3 +42,4 @@
 SHA1 (patch-bk) = 7740eea6a0d4a8ce3db3b93b2af5faacdb9424b1
 SHA1 (patch-include_pcm.h) = 1c458cd7d078356f4a970b0b3592aad139ac0918
 SHA1 (patch-src_control_control__hw.c) = 07cfe3abe586b32f6b989403cd162599da07e865
+SHA1 (patch-qq) = a41ef5695898cde72f31489de2ed10088b0f476f
diff --git a/audio/alsa-lib/patches/patch-qq b/audio/alsa-lib/patches/patch-qq
new file mode 100644
--- /dev/null
+++ b/audio/alsa-lib/patches/patch-qq
@@ -0,0 +1,72 @@
+diff -r bb67315ed01e include/global.h
+--- include/global.h	Mon May 07 21:37:17 2012 +0000
++++ include/global.h	Mon May 07 21:45:14 2012 +0000
+@@ -137,7 +137,7 @@
+ 
+ int snd_user_file(const char *file, char **result);
+ 
+-#if !defined(_POSIX_C_SOURCE) && !defined(_POSIX_SOURCE) && !defined(__NetBSD__) && !defined(__DragonFly__) && !defined(__sun)
++#if !defined(_POSIX_C_SOURCE) && !defined(_POSIX_SOURCE) && !defined(__NetBSD__) && !defined(__DragonFly__) && !defined(__sun) && !defined(_GNU_SOURCE)
+ struct timeval {
+ 	time_t		tv_sec;		/* seconds */
+ 	long		tv_usec;	/* microseconds */
+diff -r bb67315ed01e include/pcm.h
+--- include/pcm.h	Mon May 07 21:37:17 2012 +0000
++++ include/pcm.h	Mon May 07 21:45:14 2012 +0000
+@@ -928,10 +928,10 @@
+ uint32_t snd_pcm_format_silence_32(snd_pcm_format_t format);
+ uint64_t snd_pcm_format_silence_64(snd_pcm_format_t format);
+ #else
+-u_int8_t snd_pcm_format_silence(snd_pcm_format_t format);
+-u_int16_t snd_pcm_format_silence_16(snd_pcm_format_t format);
+-u_int32_t snd_pcm_format_silence_32(snd_pcm_format_t format);
+-u_int64_t snd_pcm_format_silence_64(snd_pcm_format_t format);
++uint8_t snd_pcm_format_silence(snd_pcm_format_t format);
++uint16_t snd_pcm_format_silence_16(snd_pcm_format_t format);
++uint32_t snd_pcm_format_silence_32(snd_pcm_format_t format);
++uint64_t snd_pcm_format_silence_64(snd_pcm_format_t format);
+ #endif
+ int snd_pcm_format_set_silence(snd_pcm_format_t format, void *buf, unsigned int samples);
+ 
+diff -r bb67315ed01e include/sound/asound.h
+--- include/sound/asound.h	Mon May 07 21:37:17 2012 +0000
++++ include/sound/asound.h	Mon May 07 21:45:14 2012 +0000
+@@ -358,7 +358,7 @@
+ #define SNDRV_MASK_MAX	256
+ 
+ struct sndrv_mask {
+-	u_int32_t bits[(SNDRV_MASK_MAX+31)/32];
++	uint32_t bits[(SNDRV_MASK_MAX+31)/32];
+ };
+ 
+ struct sndrv_pcm_hw_params {
+diff -r bb67315ed01e src/dlmisc.c
+--- src/dlmisc.c	Mon May 07 21:37:17 2012 +0000
++++ src/dlmisc.c	Mon May 07 21:45:14 2012 +0000
+@@ -54,11 +54,13 @@
+ #ifdef HAVE_LIBDL
+ 	if (name == NULL) {
+ 		static const char * self = NULL;
++#if 0
+ 		if (self == NULL) {
+ 			Dl_info dlinfo;
+ 			if (dladdr(snd_dlopen, &dlinfo) > 0)
+ 				self = dlinfo.dli_fname;
+ 		}
++#endif
+ 		name = self;
+ 	}
+ #endif
+diff -r fbb5dc5542a8 src/compat/hsearch_r.c
+--- src/compat/hsearch_r.c	Mon May 07 21:52:06 2012 +0000
++++ src/compat/hsearch_r.c	Mon May 07 21:52:21 2012 +0000
+@@ -1,3 +1,4 @@
++#if 0
+ /* Copyright (C) 1993, 1995, 1996, 1997 Free Software Foundation, Inc.
+    This file is part of the GNU C Library.
+    Contributed by Ulrich Drepper <drepper@gnu.ai.mit.edu>, 1993.
+@@ -234,3 +235,4 @@
+   *retval = NULL;
+   return 0;
+ }
++#endif
