# HG changeset patch
# Parent 7deb3916e9766eabd04771136f918e7a768fe95c

diff --git a/devel/libuuid/distinfo b/devel/libuuid/distinfo
--- a/devel/libuuid/distinfo
+++ b/devel/libuuid/distinfo
@@ -4,3 +4,4 @@
 RMD160 (util-linux-2.19.1.tar.bz2) = e4763d21a3a32029ccc525a0222e271f0946a4fd
 Size (util-linux-2.19.1.tar.bz2) = 4396543 bytes
 SHA1 (patch-aa) = 07d66ef414f5781d75159b446d6c822b1e543477
+SHA1 (patch-aq) = f8ededcca54e2adb13f57e564623178f4d46ced6
diff --git a/devel/libuuid/patches/patch-aq b/devel/libuuid/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/devel/libuuid/patches/patch-aq
@@ -0,0 +1,56 @@
+diff -r f3d747a80d82 shlibs/uuid/src/Makefile.am
+--- shlibs/uuid/src/Makefile.am	Sun May 06 03:03:55 2012 +0000
++++ shlibs/uuid/src/Makefile.am	Sun May 06 03:04:28 2012 +0000
+@@ -25,15 +25,15 @@
+ 
+ # move lib from $(usrlib_execdir) to $(libdir) if needed
+ install-exec-hook:
+-	if test "$(usrlib_execdir)" != "$(libdir)"; then \
+-		mkdir -p $(DESTDIR)$(libdir); \
+-		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
+-		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
+-		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
+-		(cd $(DESTDIR)$(usrlib_execdir) && \
+-			rm -f libuuid.so && \
+-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
+-	fi
++#	if test "$(usrlib_execdir)" != "$(libdir)"; then \
++#		mkdir -p $(DESTDIR)$(libdir); \
++#		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
++#		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
++#		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
++#		(cd $(DESTDIR)$(usrlib_execdir) && \
++#			rm -f libuuid.so && \
++#			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
++#	fi
+ 
+ uninstall-hook:
+ 	rm -f $(DESTDIR)$(libdir)/libuuid.so*
+diff -r f3d747a80d82 shlibs/uuid/src/Makefile.in
+--- shlibs/uuid/src/Makefile.in	Sun May 06 03:03:55 2012 +0000
++++ shlibs/uuid/src/Makefile.in	Sun May 06 03:04:28 2012 +0000
+@@ -700,15 +700,15 @@
+ 
+ # move lib from $(usrlib_execdir) to $(libdir) if needed
+ install-exec-hook:
+-	if test "$(usrlib_execdir)" != "$(libdir)"; then \
+-		mkdir -p $(DESTDIR)$(libdir); \
+-		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
+-		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
+-		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
+-		(cd $(DESTDIR)$(usrlib_execdir) && \
+-			rm -f libuuid.so && \
+-			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
+-	fi
++#	if test "$(usrlib_execdir)" != "$(libdir)"; then \
++#		mkdir -p $(DESTDIR)$(libdir); \
++#		mv $(DESTDIR)$(usrlib_execdir)/libuuid.so.* $(DESTDIR)$(libdir); \
++#		so_img_name=$$(readlink $(DESTDIR)$(usrlib_execdir)/libuuid.so); \
++#		so_img_rel_target=$$(echo $(usrlib_execdir) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
++#		(cd $(DESTDIR)$(usrlib_execdir) && \
++#			rm -f libuuid.so && \
++#			$(LN_S) $$so_img_rel_target$(libdir)/$$so_img_name libuuid.so); \
++#	fi
+ 
+ uninstall-hook:
+ 	rm -f $(DESTDIR)$(libdir)/libuuid.so*
