# HG changeset patch
# Parent ce0c2148ff7a6b9944b881ce58f1aca3340e987c

diff --git a/devel/nss/Makefile b/devel/nss/Makefile
--- a/devel/nss/Makefile
+++ b/devel/nss/Makefile
@@ -48,6 +48,17 @@
 MAKE_ENV+=	NSS_USE_SYSTEM_SQLITE=1
 MAKE_ENV+=	EXTRA_SHARED_LIBS="-Wl,-R${BUILDLINK_PREFIX.nspr}/lib/nspr -Wl,-R,${PREFIX}/lib/nss -L${BUILDLINK_PREFIX.nspr}/lib/nspr -lplc4 -lplds4 -lnspr4"
 
+# [Gregor] And just where is this type SUPPOSED to come from?
+_WRAP_EXTRA_ARGS.CC+=-Duint=uint32_t
+
+# [Gregor] This build loses, but needs, our CFLAGS
+_WRAP_EXTRA_ARGS.CC+=-D_GNU_SOURCE -D_BSD_SOURCE
+
+# [Gregor] More misdetection
+pre-build:
+	${SED} 's/DHAVE_SYS_CDEFS_H/UHAVE_SYS_CDEFS_H/' -i ${WRKSRC}/security/dbm/config/config.mk
+	${SED} 's/(ffs/(__builtin_ffs/g' -i ${WRKSRC}/security/nss/cmd/checkcert/checkcert.c
+
 pre-configure:
 # sanity check: make sure the nss release is what we think it is.
 #	@set -e; if [ '"${NSS_RELEASE}"' !=			\
