# HG changeset patch
# Parent 5112e02d66914f5dcc250c58de6754fe40bafffe

diff --git a/devel/imake/Makefile b/devel/imake/Makefile
--- a/devel/imake/Makefile
+++ b/devel/imake/Makefile
@@ -42,6 +42,9 @@
 SUBST_FILES.xmkmf=	xmkmf
 SUBST_SED.xmkmf=	-e "s/^configdirspec= */configdirspec=/"
 
+pre-build:
+	sed 's/-x c -/-x c - -x none/' -i ${WRKSRC}/imake.c
+
 .include "../../x11/xproto/buildlink3.mk"
 
 .include "../../mk/bsd.pkg.mk"
