# HG changeset patch
# Parent 2397f135673dc32495ea57739361f166da21fc0b

diff --git a/devel/pcre/Makefile b/devel/pcre/Makefile
--- a/devel/pcre/Makefile
+++ b/devel/pcre/Makefile
@@ -21,6 +21,8 @@
 CONFIGURE_ARGS+=	--enable-utf8 --enable-unicode-properties
 CONFIGURE_ARGS+=	--enable-pcre16
 
+CONFIGURE_ARGS+=	--disable-cpp
+
 SUBST_CLASSES+=		rpath
 SUBST_STAGE.rpath=	pre-configure
 SUBST_MESSAGE.rpath=	Fixing rpath output in pcre-config
diff --git a/devel/pcre/PLIST b/devel/pcre/PLIST
--- a/devel/pcre/PLIST
+++ b/devel/pcre/PLIST
@@ -3,18 +3,12 @@
 bin/pcregrep
 bin/pcretest
 include/pcre.h
-include/pcre_scanner.h
-include/pcre_stringpiece.h
-include/pcrecpp.h
-include/pcrecpparg.h
 include/pcreposix.h
 lib/libpcre.la
 lib/libpcre16.la
-lib/libpcrecpp.la
 lib/libpcreposix.la
 lib/pkgconfig/libpcre.pc
 lib/pkgconfig/libpcre16.pc
-lib/pkgconfig/libpcrecpp.pc
 lib/pkgconfig/libpcreposix.pc
 man/man1/pcre-config.1
 man/man1/pcregrep.1
@@ -75,7 +69,6 @@
 man/man3/pcrebuild.3
 man/man3/pcrecallout.3
 man/man3/pcrecompat.3
-man/man3/pcrecpp.3
 man/man3/pcrejit.3
 man/man3/pcrelimits.3
 man/man3/pcrematching.3
@@ -127,7 +120,6 @@
 share/doc/pcre/html/pcrebuild.html
 share/doc/pcre/html/pcrecallout.html
 share/doc/pcre/html/pcrecompat.html
-share/doc/pcre/html/pcrecpp.html
 share/doc/pcre/html/pcredemo.html
 share/doc/pcre/html/pcregrep.html
 share/doc/pcre/html/pcrejit.html
