# HG changeset patch
# Parent 634363e5e84e930158106aef8415bf509c22707b

diff --git a/devel/nspr/Makefile b/devel/nspr/Makefile
--- a/devel/nspr/Makefile
+++ b/devel/nspr/Makefile
@@ -72,6 +72,13 @@
 BUILDLINK_PASSTHRU_DIRS+=	Developer/Headers/FlatCarbon
 .endif
 
+# [Gregor] SOL_TCP is deprecated
+CFLAGS+=-DSOL_TCP=IPPROTO_TCP
+
+# [Gregor] the detection is broken
+pre-build:
+	${SED} 's/#define _PR_HAVE_GETPROTO_R.*//' -i ${WRKSRC}/nsprpub/pr/src/misc/prnetdb.c
+
 pre-configure:
 	cd ${WRKSRC}/nsprpub && autoconf
 
