# HG changeset patch
# Parent a6cda5d01bb9093381f768af6f3cdfef776e6164

diff --git a/devel/nss/Makefile b/devel/nss/Makefile
--- a/devel/nss/Makefile
+++ b/devel/nss/Makefile
@@ -50,6 +50,17 @@
 MAKE_ENV+=	NSS_USE_SYSTEM_SQLITE=1
 MAKE_ENV+=	EXTRA_SHARED_LIBS="-Wl,-R${BUILDLINK_PREFIX.nspr}/lib/nspr -Wl,-R,${PREFIX}/lib/nss -L${BUILDLINK_PREFIX.nspr}/lib/nspr -lplc4 -lplds4 -lnspr4"
 
+# [Gregor] And just where is this type SUPPOSED to come from?
+_WRAP_EXTRA_ARGS.CC+=-Duint=uint32_t
+
+# [Gregor] This build loses, but needs, our CFLAGS
+_WRAP_EXTRA_ARGS.CC+=-D_GNU_SOURCE -D_BSD_SOURCE
+
+# [Gregor] More misdetection
+pre-build:
+	${SED} 's/DHAVE_SYS_CDEFS_H/UHAVE_SYS_CDEFS_H/' -i ${WRKSRC}/security/dbm/config/config.mk
+	${SED} 's/(ffs/(__builtin_ffs/g' -i ${WRKSRC}/security/nss/cmd/checkcert/checkcert.c
+
 do-build:
 	cd ${WRKSRC}/security/nss && ${SETENV} ${MAKE_ENV} ${MAKE_PROGRAM} \
 	  build_coreconf build_dbm all
