# HG changeset patch
# Parent 895d00d78a518bd34177ffe562b9f1bbdb9f3e85

diff --git a/devel/flex/PLIST b/devel/flex/PLIST
--- a/devel/flex/PLIST
+++ b/devel/flex/PLIST
@@ -6,19 +6,3 @@
 lib/libfl.a
 lib/libfl_pic.a
 man/man1/flex.1
-share/locale/ca/LC_MESSAGES/flex.mo
-share/locale/da/LC_MESSAGES/flex.mo
-share/locale/de/LC_MESSAGES/flex.mo
-share/locale/es/LC_MESSAGES/flex.mo
-share/locale/fr/LC_MESSAGES/flex.mo
-share/locale/ga/LC_MESSAGES/flex.mo
-share/locale/ko/LC_MESSAGES/flex.mo
-share/locale/nl/LC_MESSAGES/flex.mo
-share/locale/pl/LC_MESSAGES/flex.mo
-share/locale/pt_BR/LC_MESSAGES/flex.mo
-share/locale/ro/LC_MESSAGES/flex.mo
-share/locale/ru/LC_MESSAGES/flex.mo
-share/locale/sv/LC_MESSAGES/flex.mo
-share/locale/tr/LC_MESSAGES/flex.mo
-share/locale/vi/LC_MESSAGES/flex.mo
-share/locale/zh_CN/LC_MESSAGES/flex.mo
