# HG changeset patch
# Parent f53913e5268aba4ee1e23204d15dcdf0b2e6a14a

diff --git a/devel/glib2/Makefile b/devel/glib2/Makefile
--- a/devel/glib2/Makefile
+++ b/devel/glib2/Makefile
@@ -48,6 +48,12 @@
 .include "../../devel/libelf/buildlink3.mk"
 .endif
 
+
+
+# [Gregor] bugs in how it uses libdl
+pre-build:
+	sed -i 's/if (msg)/if (!p \&\& msg)/' ${WRKSRC}/gmodule/gmodule-dl.c
+
 .include "../../converters/libiconv/buildlink3.mk"
 .include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../devel/pcre/buildlink3.mk"
