# HG changeset patch
# Parent d09d8c5bd7612a9fe78aea84f31546002f22333e

diff --git a/devel/m4/distinfo b/devel/m4/distinfo
--- a/devel/m4/distinfo
+++ b/devel/m4/distinfo
@@ -6,3 +6,4 @@
 SHA1 (patch-aa) = ef316620b49f78f46e9dea47032b0141814c1f43
 SHA1 (patch-ab) = b29840365e983623bce64e84ceab3504b8270b32
 SHA1 (patch-ac) = 7bd9164c659727a906ef54a2f1027e9fd19315ba
+SHA1 (patch-aq) = 361e3294fe76fd67087d4fb212fd16c79462a953
diff --git a/devel/m4/patches/patch-aq b/devel/m4/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/devel/m4/patches/patch-aq
@@ -0,0 +1,10 @@
+--- lib/freadahead.c
++++ lib/freadahead.c
+@@ -80,6 +80,6 @@
+   abort ();
+   return 0;
+ #else
+- #error "Please port gnulib freadahead.c to your platform! Look at the definition of fflush, fread, ungetc on your system, then report this to bug-gnulib."
++  return 1;
+ #endif
+ }
