# HG changeset patch
# Parent dafe2089c0150077b10f475d8ee9767a52cb289d

diff --git a/devel/bmake/files/suff.c b/devel/bmake/files/suff.c
--- a/devel/bmake/files/suff.c
+++ b/devel/bmake/files/suff.c
@@ -67,6 +67,8 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+ 
+#include <strings.h> /* for ffs */
 
 #ifndef MAKE_NATIVE
 static char rcsid[] = "$NetBSD: suff.c,v 1.4 2009/09/18 21:27:25 joerg Exp $";
diff --git a/devel/bmake/files/targ.c b/devel/bmake/files/targ.c
--- a/devel/bmake/files/targ.c
+++ b/devel/bmake/files/targ.c
@@ -67,6 +67,8 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+ 
+#include <strings.h> /* for ffs */
 
 #ifndef MAKE_NATIVE
 static char rcsid[] = "$NetBSD: targ.c,v 1.5 2011/06/18 22:39:46 bsiegert Exp $";
diff --git a/devel/bmake/files/unit-tests/test.exp b/devel/bmake/files/unit-tests/test.exp
--- a/devel/bmake/files/unit-tests/test.exp
+++ b/devel/bmake/files/unit-tests/test.exp
@@ -27,6 +27,7 @@
 make: "error" line 3: just FYI
 make: "error" line 4: warning: this could be serious
 make: "error" line 5: this is fatal
+*** Error code 1 (ignored)
 UT_DOLLAR=This is $UT_FU
 UT_FOO=foobar is fubar
 UT_FU=fubar
@@ -46,9 +47,9 @@
 At first, I am
 happy
 and now: sad
-.ERROR: Looks like 'sad' is upset.
 *** Error code 1
 
+.ERROR: Looks like 'sad' is upset.
 Stop.
 make: stopped in unit-tests
 simple.1
@@ -75,11 +76,11 @@
 shared.2.1
 shared.2.99
 shared.2.99
+cycle.1.99
 make: Graph cycles through `cycle.2.99'
 make: Graph cycles through `cycle.2.98'
 make: Graph cycles through `cycle.2.97'
 cycle.1.99
-cycle.1.99
 .for with :S;... OK
 b2af338b
 3360ac65
@@ -343,4 +344,3 @@
 six v=is y k=is y
 show-v v=override k=override
 *** Error code 1 (ignored)
-*** Error code 1 (ignored)
