# HG changeset patch
# Parent 15277a333855d54fb7646f24136f1c23369d999c

diff --git a/devel/bison/Makefile b/devel/bison/Makefile
--- a/devel/bison/Makefile
+++ b/devel/bison/Makefile
@@ -27,6 +27,7 @@
 # Avoid rebuilding manpage
 pre-build:
 	${TOUCH} ${WRKSRC}/doc/bison.1
+	sed -i 's/#elif 0/#elif 1/' -i ${WRKSRC}/lib/fseterr.c
 
 # "bison" wants a recent version of "gettext" which at least some
 # NetBSD versions don't provide. Figure out whether it will install
