# HG changeset patch
# Parent e684c6ce703b5060b97194accbff8067299e3cea
diff --git a/devel/libowfat/distinfo b/devel/libowfat/distinfo
--- a/devel/libowfat/distinfo
+++ b/devel/libowfat/distinfo
@@ -4,3 +4,4 @@
 RMD160 (libowfat-0.28.tar.bz2) = 3e8826e10a3f46520bdfa9602edb2741728aa370
 Size (libowfat-0.28.tar.bz2) = 129622 bytes
 SHA1 (patch-aa) = 7542a1fd6b2f4f3f1dd430be547075e98488d7d0
+SHA1 (patch-aq) = 15fefa79ec4f15d900ec1e1a97599ed5dd524b16
diff --git a/devel/libowfat/patches/patch-aq b/devel/libowfat/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/devel/libowfat/patches/patch-aq
@@ -0,0 +1,11 @@
+diff -r 88c77b0efab8 GNUmakefile
+--- GNUmakefile	Fri Jul 20 02:17:36 2012 +0000
++++ GNUmakefile	Fri Jul 20 02:19:53 2012 +0000
+@@ -24,6 +24,7 @@
+ # CFLAGS += -fstrict-aliasing -Wstrict-aliasing=2
+ 
+ CFLAGS += -D_REENTRANT
++CFLAGS += -D_GNU_SOURCE
+ 
+ # startrip
+ ifneq ($(DEBUG),)
