# HG changeset patch
# Parent c3338fed0d2d920567a16cc93fcde01aea037865

diff --git a/editors/vim-share/Makefile.common b/editors/vim-share/Makefile.common
--- a/editors/vim-share/Makefile.common
+++ b/editors/vim-share/Makefile.common
@@ -86,5 +86,8 @@
 CONFIGURE_ARGS+=	--enable-rubyinterp
 .endif
 
+# [Gregor] vim doesn't recognize that it needs curses to build ANYTHING, even the share component
+.include "../../mk/curses.buildlink3.mk"
+
 post-patch:
 	find ${WRKDIR} -name '*.orig_dist' -exec rm -f {} \;
