# HG changeset patch
# Parent cc3a140f877cac32b7593649fe61546dbef3f5c0

diff --git a/graphics/MesaLib/Makefile b/graphics/MesaLib/Makefile
--- a/graphics/MesaLib/Makefile
+++ b/graphics/MesaLib/Makefile
@@ -72,6 +72,7 @@
 
 pre-build:
 	cd ${WRKSRC} && ${RM} -fr src/glu src/glut progs
+	sed -i 's/#if defined(__linux__) && defined(__i386__)/#if 0/' ${WRKSRC}/src/mesa/main/glheader.h
 
 .include "../../mk/compiler.mk"
 
