# HG changeset patch
# Parent 95f70499a05eb4a5cd815ac4d4e4e972985e49d0

diff --git a/graphics/jpeg/buildlink3.mk b/graphics/jpeg/buildlink3.mk
--- a/graphics/jpeg/buildlink3.mk
+++ b/graphics/jpeg/buildlink3.mk
@@ -5,9 +5,10 @@
 .if !defined(JPEG_BUILDLINK3_MK)
 JPEG_BUILDLINK3_MK:=
 
-BUILDLINK_API_DEPENDS.jpeg+=	jpeg>=8nb1
-#BUILDLINK_ABI_DEPENDS.jpeg+=	jpeg>=8nb1
-BUILDLINK_ABI_DEPENDS.jpeg+=	jpeg>=8nb1
+# [Gregor] Reduced to 8 due to version diffs between pkgresolve and pkgsrc
+BUILDLINK_API_DEPENDS.jpeg+=	jpeg>=8
+#BUILDLINK_ABI_DEPENDS.jpeg+=	jpeg>=8
+BUILDLINK_ABI_DEPENDS.jpeg+=	jpeg>=8
 BUILDLINK_PKGSRCDIR.jpeg?=	../../graphics/jpeg
 .endif # JPEG_BUILDLINK3_MK
 
