# HG changeset patch
# Parent 2cff2e7332bfa99196cdf4173c96460523b7b1b6

diff --git a/graphics/gimp/PLIST b/graphics/gimp/PLIST
--- a/graphics/gimp/PLIST
+++ b/graphics/gimp/PLIST
@@ -316,7 +316,6 @@
 lib/gimp/2.0/plug-ins/file-tiff-load
 lib/gimp/2.0/plug-ins/file-tiff-save
 lib/gimp/2.0/plug-ins/file-uri
-lib/gimp/2.0/plug-ins/file-wmf
 lib/gimp/2.0/plug-ins/file-xbm
 lib/gimp/2.0/plug-ins/file-xjt
 ${PLIST.remote}lib/gimp/2.0/plug-ins/file-xmc
