# HG changeset patch
# Parent eeb3875686336028a0eee5c8533353defeeca1b1

diff --git a/graphics/aalib/distinfo b/graphics/aalib/distinfo
--- a/graphics/aalib/distinfo
+++ b/graphics/aalib/distinfo
@@ -6,3 +6,4 @@
 SHA1 (patch-aa) = 927d69a8f499b99f0bd5f13cf220ce8cbb7ea591
 SHA1 (patch-ab) = 272f6f10dbe7a525c582a9b4c865f0ab9b2f9fba
 SHA1 (patch-ac) = 8d9c812b9b871f44dd1285d6a635eeab6e49bf49
+SHA1 (patch-aq) = 285c28a811719d0e83f8df29c3a6bf93fb3eef8f
diff --git a/graphics/aalib/patches/patch-aq b/graphics/aalib/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/graphics/aalib/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 6fe3c134e687 src/aalinuxkbd.c
+--- src/aalinuxkbd.c	Mon May 07 17:51:38 2012 +0000
++++ src/aalinuxkbd.c	Mon May 07 17:51:56 2012 +0000
+@@ -346,7 +346,7 @@
+ #endif
+ static char sig2catch[] =
+ {SIGHUP, SIGINT, SIGQUIT, SIGILL,
+- SIGTRAP, SIGIOT, SIGBUS, SIGFPE,
++ SIGTRAP, SIGABRT, SIGBUS, SIGFPE,
+  SIGSEGV, SIGPIPE, SIGALRM, SIGTERM,
+  SIGXCPU, SIGXFSZ, SIGVTALRM,
+ 	   /* SIGPROF , */ SIGPWR};
