# HG changeset patch
# Parent 60e5cea5a387ac475fb843a377980920f0df3987

diff --git a/lang/python26/Makefile b/lang/python26/Makefile
--- a/lang/python26/Makefile
+++ b/lang/python26/Makefile
@@ -117,22 +117,7 @@
 .endif
 
 PLIST_VARS+=	dll nis no-nis
-.if ${OPSYS} == "IRIX"
-.  if ${ABI} == "64"
-PLIST.no-nis=	yes
-.  else
-PLIST.nis=	yes
-.  endif
-.else
 PLIST.dll=	yes
-.  if ${OPSYS} == "MirBSD"
-# neither nis nor no-nis
-.  elif ${OPSYS} != "NetBSD" || exists(/usr/bin/ypcat)
-PLIST.nis=	yes
-.  else
-PLIST.no-nis=	yes
-.  endif
-.endif
 
 .include "../../mk/bdb.buildlink3.mk"
 MAKE_ENV+=	PY_BDB_TYPE=${BDB_TYPE}
