# HG changeset patch
# Parent f9c0cb325932319928fe20b7a8abed7a2bc61954

diff --git a/lang/python27/PLIST.common b/lang/python27/PLIST.common
--- a/lang/python27/PLIST.common
+++ b/lang/python27/PLIST.common
@@ -1836,7 +1836,6 @@
 lib/python${PY_VER_SUFFIX}/lib-dynload/itertools.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/math.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/mmap.so
-${PLIST.nis}lib/python${PY_VER_SUFFIX}/lib-dynload/nis.so
 ${PLIST.no-nis}lib/python${PY_VER_SUFFIX}/lib-dynload/nis_failed.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/operator.so
 lib/python${PY_VER_SUFFIX}/lib-dynload/parser.so
