# HG changeset patch
# Parent 327d15c0349594f705e342ed2d6ecc613baecfc1

diff --git a/misc/pty-redir/distinfo b/misc/pty-redir/distinfo
--- a/misc/pty-redir/distinfo
+++ b/misc/pty-redir/distinfo
@@ -4,3 +4,4 @@
 RMD160 (pty-redir-0.1.tar.gz) = 75347881746dd1419e95e2ada30b23b2d9c0d223
 Size (pty-redir-0.1.tar.gz) = 8405 bytes
 SHA1 (patch-aa) = 28f8a1c22a9bf602e0e447ca45736e7cde6fddd5
+SHA1 (patch-aq) = 17bdbfdef55312079c5ab32069556b50658dd771
diff --git a/misc/pty-redir/patches/patch-aq b/misc/pty-redir/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/misc/pty-redir/patches/patch-aq
@@ -0,0 +1,51 @@
+--- pty-redir.c
++++ pty-redir.c
+@@ -43,30 +43,18 @@
+ #include <fcntl.h>
+ 
+ 
+-#define PTY00 "/dev/ptyXX"
+-#define PTY10 "pqrs"
+-#define PTY01 "0123456789abcdef"
++#define PTY00 "/dev/ptmx"
+ 
+ static int
+ getPtyMaster(char *tty10, char *tty01)
+ {
+-    char *p10;
+-    char *p01;
+     static char dev[] = PTY00;
+     int fd;
+ 
+-    for (p10 = PTY10; *p10 != '\0'; p10++) {
+-        dev[8] = *p10;
+-        for (p01 = PTY01; *p01 != '\0'; p01++) {
+-            dev[9] = *p01;
+             fd = open(dev, O_RDWR);
+             if (fd >= 0) {
+-                *tty10 = *p10;
+-                *tty01 = *p01;
+                 return fd;
+             }
+-        }
+-    }
+     fprintf(stderr,"Ran out of pty.\n");
+     exit(1);
+     return fd;
+@@ -81,8 +69,6 @@
+ 	int i;
+ 	
+ 	fd=getPtyMaster(&a,&b);
+-	fprintf(stdout,"/dev/tty%c%c",a,b);
+-	fflush(stdout);
+ 	if(NULL==(args=malloc((argc)*sizeof(char *))))
+ 	{
+ 		fprintf(stderr,"Virtual memory exhausted\n");
+@@ -95,7 +81,6 @@
+ 	args[argc-1]=NULL; /*End of params*/
+ 	dup2(fd,0);
+ 	dup2(fd,1);
+-	fflush(NULL);
+ 	if(fork())
+ 	{
+ 		exit(0);
