# HG changeset patch
# Parent 6346450642750ff26f935e8c788d3a6673772d20

diff --git a/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk b/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk
--- a/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk
+++ b/pkgtools/bootstrap-mk-files/files/mods/Linux.bsd.man.mk
@@ -13,6 +13,7 @@
 .endif
 
 .PHONY:		catinstall maninstall catpages manpages catlinks manlinks cleanman html installhtml cleanhtml
+MKMAN?=no
 .if ${MKMAN} != "no"
 realinstall:	${MANINSTALL}
 .endif
@@ -40,11 +41,10 @@
 TMACDIR?=	${DESTDIR}/usr/share/groff/tmac
 .endif
 HTMLDIR?=	${DESTDIR}/usr/share/man
-CATDEPS?=	${TMACDIR}/mandoc.tmac \
-		${TMACDIR}/doc.tmac
+CATDEPS?=
 MANTARGET?=	cat
-NROFF?=		nroff -Tascii
-GROFF?=		groff -Tascii
+NROFF?=		cat
+GROFF?=		cat
 TBL?=		tbl
 
 
