# HG changeset patch
# Parent 2947b0ad91a38b966d1a0f2d2ac14a793bb63180

diff --git a/pkgtools/pkg_install-info/files/install-info.c b/pkgtools/pkg_install-info/files/install-info.c
--- a/pkgtools/pkg_install-info/files/install-info.c
+++ b/pkgtools/pkg_install-info/files/install-info.c
@@ -21,6 +21,8 @@
 #include "system.h"
 #include <getopt.h>
 
+#include <stdlib.h>
+
 static char *progname = "install-info";
 
 int menu_line_equal (), menu_line_lessp ();
diff --git a/pkgtools/pkg_install-info/files/system.h b/pkgtools/pkg_install-info/files/system.h
--- a/pkgtools/pkg_install-info/files/system.h
+++ b/pkgtools/pkg_install-info/files/system.h
@@ -21,7 +21,9 @@
 #ifndef TEXINFO_SYSTEM_H
 #define TEXINFO_SYSTEM_H
 
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
 
 #include <config.h>
 
diff --git a/pkgtools/pkg_install-info/files/xexit.c b/pkgtools/pkg_install-info/files/xexit.c
--- a/pkgtools/pkg_install-info/files/xexit.c
+++ b/pkgtools/pkg_install-info/files/xexit.c
@@ -19,6 +19,8 @@
 
 #include "system.h"
 
+#include <stdlib.h>
+
 /* SunOS 4.1.1 gets STDC_HEADERS defined, but it doesn't provide
    EXIT_FAILURE.  So far no system has defined one of EXIT_FAILURE and
    EXIT_SUCCESS without the other.  */
