# HG changeset patch
# Parent f92f56a56b0e66e918e877584c94eaf34238d9a0

diff --git a/security/p5-Crypt-Rijndael/distinfo b/security/p5-Crypt-Rijndael/distinfo
--- a/security/p5-Crypt-Rijndael/distinfo
+++ b/security/p5-Crypt-Rijndael/distinfo
@@ -3,3 +3,4 @@
 SHA1 (Crypt-Rijndael-1.09.tar.gz) = 2edc7836a62c01c6db668586f907f8f57170a970
 RMD160 (Crypt-Rijndael-1.09.tar.gz) = a0fa349631ce7a5555a3f6a9fe6c23b1a745858d
 Size (Crypt-Rijndael-1.09.tar.gz) = 28994 bytes
+SHA1 (patch-aq) = 4780039094f3df9bde4617f4f26c11a6f23b56bf
diff --git a/security/p5-Crypt-Rijndael/patches/patch-aq b/security/p5-Crypt-Rijndael/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/security/p5-Crypt-Rijndael/patches/patch-aq
@@ -0,0 +1,14 @@
+diff -r d81bee0804d7 rijndael.h
+--- rijndael.h	Mon Jul 02 16:04:39 2012 +0000
++++ rijndael.h	Mon Jul 02 16:05:26 2012 +0000
+@@ -74,8 +74,8 @@
+ /* I expect this to be the usual case */
+ #if ! defined(_CRYPT_RIJNDAEL_H_TYPES) && ( defined(_SYS_TYPES_H) || defined(_SYS_TYPES_H_) )   
+ 	#define _CRYPT_RIJNDAEL_H_TYPES
+-	typedef __uint32_t UINT32;
+-	typedef __uint8_t  UINT8;
++	typedef uint32_t UINT32;
++	typedef uint8_t  UINT8;
+ #endif
+ 
+ #if defined(__CYGWIN__) && ! defined(_CRYPT_RIJNDAEL_H_TYPES)
