# HG changeset patch
# Parent 26f21c127356a29616605b9d868e87a441f322e4

diff --git a/security/libgpg-error/PLIST b/security/libgpg-error/PLIST
--- a/security/libgpg-error/PLIST
+++ b/security/libgpg-error/PLIST
@@ -8,13 +8,3 @@
 share/common-lisp/source/gpg-error/gpg-error-package.lisp
 share/common-lisp/source/gpg-error/gpg-error.asd
 share/common-lisp/source/gpg-error/gpg-error.lisp
-share/locale/cs/LC_MESSAGES/libgpg-error.mo
-share/locale/de/LC_MESSAGES/libgpg-error.mo
-share/locale/fr/LC_MESSAGES/libgpg-error.mo
-share/locale/it/LC_MESSAGES/libgpg-error.mo
-share/locale/nl/LC_MESSAGES/libgpg-error.mo
-share/locale/pl/LC_MESSAGES/libgpg-error.mo
-share/locale/ro/LC_MESSAGES/libgpg-error.mo
-share/locale/sv/LC_MESSAGES/libgpg-error.mo
-share/locale/vi/LC_MESSAGES/libgpg-error.mo
-share/locale/zh_CN/LC_MESSAGES/libgpg-error.mo
