# HG changeset patch
# Parent 1a8bd8e34ee816c3623935ed38cfa3215e5524e0

diff --git a/security/openssl/Makefile b/security/openssl/Makefile
--- a/security/openssl/Makefile
+++ b/security/openssl/Makefile
@@ -61,6 +61,7 @@
 # back to ./config's autodetection if not.
 .  if defined(OPENSSL_MACHINE_ARCH) && !empty(OPENSSL_MACHINE_ARCH)
 CONFIGURE_SCRIPT=	./Configure
+CONFIGURE_ARGS+=	no-shared no-dso zlib enable-md2 -Wa,--noexecstack
 .    if !empty(CC_VERSION:Mgcc*)
 CONFIGURE_ARGS+=	solaris${${ABI}==64:?64:}-${OPENSSL_MACHINE_ARCH}-gcc
 .    else
@@ -154,8 +155,16 @@
 
 INSTALLATION_DIRS+=	share/examples/openssl
 
+pre-build:
+	touch ${WRKSRC}/include/memory.h
+	cd ${WRKSRC} && $(MAKE) depend
+
 # Fix the path to perl in various scripts.
 pre-configure:
 	cd ${WRKSRC} && ${PERL5} util/perlpath.pl ${PERL5}
-
+	sed -i '/^"linux-x86_64/s/-DTERMIO/-DTERMIOS/' ${WRKSRC}/Configure
+	sed -i '/^"linux-elf/s/-DTERMIO/-DTERMIOS/' ${WRKSRC}/Configure
+	sed -i 's/defined(linux)/0/' ${WRKSRC}/crypto/ui/ui_openssl.c
+	sed -i 's/\!defined(_UNICOS)/0/' ${WRKSRC}/apps/speed.c
+	
 .include "../../mk/bsd.pkg.mk"
