# HG changeset patch
# Parent 231063f30af5c98fea56e9c0ff46450a544a4f9e

diff --git a/shells/bash/Makefile b/shells/bash/Makefile
--- a/shells/bash/Makefile
+++ b/shells/bash/Makefile
@@ -78,6 +78,9 @@
 .include "../../devel/readline/buildlink3.mk"
 .endif
 
+# [Gregor] mischeck for musl, safe everywhere
+CONFIGURE_ARGS+=		--without-bash-malloc
+
 post-install:
 	${INSTALL_MAN} ${WRKSRC}/doc/bash.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
 	${INSTALL_MAN} ${WRKSRC}/doc/bashbug.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
