# HG changeset patch
# Parent 5b5a633539b34b6534ad5bd6fb2456e5d1862924

diff --git a/sysutils/libpciaccess/distinfo b/sysutils/libpciaccess/distinfo
--- a/sysutils/libpciaccess/distinfo
+++ b/sysutils/libpciaccess/distinfo
@@ -5,3 +5,4 @@
 Size (libpciaccess-0.11.0.tar.bz2) = 295639 bytes
 SHA1 (patch-aa) = 6bb844c0696c8a86c6033605339e82da34164bdc
 SHA1 (patch-ac) = c2df07d306203f3236e3a79caf4eedc5104d6f78
+SHA1 (patch-aq) = 0f0850644192c53cab4e5a8105e6920ff46e8171
diff --git a/sysutils/libpciaccess/patches/patch-aq b/sysutils/libpciaccess/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/libpciaccess/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 6157f7aaf390 src/linux_sysfs.c
+--- src/linux_sysfs.c	Thu Jul 19 02:37:38 2012 +0000
++++ src/linux_sysfs.c	Thu Jul 19 02:38:32 2012 +0000
+@@ -44,6 +44,8 @@
+ #include <dirent.h>
+ #include <errno.h>
+ 
++#include <limits.h>
++
+ #include "config.h"
+ 
+ #ifdef HAVE_MTRR
