# HG changeset patch
# Parent e596af157328b73e930c2c62346e572576f21e7d

diff --git a/sysutils/pciutils/distinfo b/sysutils/pciutils/distinfo
--- a/sysutils/pciutils/distinfo
+++ b/sysutils/pciutils/distinfo
@@ -7,3 +7,4 @@
 SHA1 (patch-ab) = 1448bb418cdc908eb6652cc98994d23d0d78cea0
 SHA1 (patch-ac) = 4b90e99c051391b9cd3a6c1f5781f41b4a855c29
 SHA1 (patch-ad) = bfe9069e96fd46798afec73c93ad71db3a0e0e37
+SHA1 (patch-aq) = 5546296816722c621aeb777ccd6f316c04173182
diff --git a/sysutils/pciutils/patches/patch-aq b/sysutils/pciutils/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/pciutils/patches/patch-aq
@@ -0,0 +1,26 @@
+diff -r 285cb9c34d4e lib/configure
+--- lib/configure	Fri Jul 20 01:48:40 2012 +0000
++++ lib/configure	Fri Jul 20 01:49:39 2012 +0000
+@@ -68,7 +68,7 @@
+ 		echo >>$c '#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"'
+ 		echo >>$c '#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"'
+ 		case $cpu in
+-				i?86|x86_64)	echo_n " i386-ports"
++				disabled-i?86|disabled-x86_64)	echo_n " i386-ports"
+ 						echo >>$c '#define PCI_HAVE_PM_INTEL_CONF'
+ 						;;
+ 		esac
+diff -r 285cb9c34d4e lib/names-net.c
+--- lib/names-net.c	Fri Jul 20 01:48:40 2012 +0000
++++ lib/names-net.c	Fri Jul 20 01:49:39 2012 +0000
+@@ -5,6 +5,10 @@
+  *
+  *	Can be freely distributed and used under the terms of the GNU GPL.
+  */
++ 
++#ifndef _GNU_SOURCE
++#define _GNU_SOURCE
++#endif
+ 
+ #include <string.h>
+ #include <stdlib.h>
