# HG changeset patch
# Parent 428060a92787d438dbd01ca00f6192cf35c6a277

diff --git a/sysutils/gamin/distinfo b/sysutils/gamin/distinfo
--- a/sysutils/gamin/distinfo
+++ b/sysutils/gamin/distinfo
@@ -16,3 +16,4 @@
 SHA1 (patch-server_gam__node.h) = 49a0d381b4900564c4f0ba7a654eb22e294c0a16
 SHA1 (patch-server_gam__subscription.c) = 58f6c10e4e1e899d097662536ea31bf8a692b114
 SHA1 (patch-server_gam__subscription.h) = 6347865210aa0338bc8b0a81fc671824c786fd26
+SHA1 (patch-aq) = c3d7a1f7ae26f6b6bfb9b80c1586bf9f9f70685e
diff --git a/sysutils/gamin/patches/patch-aq b/sysutils/gamin/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/gamin/patches/patch-aq
@@ -0,0 +1,17 @@
+diff -r c02b16f5e18d server/gam_dnotify.c
+--- server/gam_dnotify.c	Fri Jul 20 00:43:51 2012 +0000
++++ server/gam_dnotify.c	Fri Jul 20 00:46:15 2012 +0000
+@@ -19,7 +19,13 @@
+ 
+ #include "server_config.h"
+ #define _GNU_SOURCE
++#if defined(__linux__) && !defined(__GLIBC__)
++/* need to do this nasty hack until/unless musl exposes the right #defines */
++#define HAVE_ARCH_STRUCT_FLOCK
++#include <linux/fcntl.h>
++#else
+ #include <fcntl.h>
++#endif
+ #include <signal.h>
+ #include <unistd.h>
+ #include <stdio.h>
