# HG changeset patch
# Parent 845e835d6cfc54e889f9403cac9416a77db455a5

diff --git a/sysutils/pciutils/distinfo b/sysutils/pciutils/distinfo
--- a/sysutils/pciutils/distinfo
+++ b/sysutils/pciutils/distinfo
@@ -7,3 +7,4 @@
 SHA1 (patch-ab) = 1448bb418cdc908eb6652cc98994d23d0d78cea0
 SHA1 (patch-ac) = c917e8ad2d1e87fa65a2d8f3256afd40d5a87396
 SHA1 (patch-ad) = bfe9069e96fd46798afec73c93ad71db3a0e0e37
+SHA1 (patch-aq) = b29a7e176a23456b6c509e5c40326d5a59050b8c
diff --git a/sysutils/pciutils/patches/patch-aq b/sysutils/pciutils/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/pciutils/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 63e97996f76b lib/configure
+--- lib/configure	Sun May 06 02:42:29 2012 +0000
++++ lib/configure	Sun May 06 02:42:58 2012 +0000
+@@ -68,7 +68,7 @@
+ 		echo >>$c '#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"'
+ 		echo >>$c '#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"'
+ 		case $cpu in
+-				i?86|x86_64)	echo_n " i386-ports"
++				disabled-i?86|disabled-x86_64)	echo_n " i386-ports"
+ 						echo >>$c '#define PCI_HAVE_PM_INTEL_CONF'
+ 						;;
+ 		esac
