# HG changeset patch
# Parent 479fec007879a729bb23f8654fea7916eac470d4

diff --git a/sysutils/coreutils/Makefile b/sysutils/coreutils/Makefile
--- a/sysutils/coreutils/Makefile
+++ b/sysutils/coreutils/Makefile
@@ -61,6 +61,8 @@
 #MESSAGE_SUBST+=	OPSYS=${OPSYS:Q}
 #.endif
 
+CFLAGS+=		-DSLOW_BUT_NO_HACKS
+
 .include "../../converters/libiconv/buildlink3.mk"
 .include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
diff --git a/sysutils/coreutils/distinfo b/sysutils/coreutils/distinfo
--- a/sysutils/coreutils/distinfo
+++ b/sysutils/coreutils/distinfo
@@ -5,3 +5,4 @@
 Size (coreutils-8.13.tar.gz) = 11660808 bytes
 SHA1 (patch-lib-fflush.c) = e52e0da2127c5debbb14e967b1e206226efdaa4a
 SHA1 (patch-lib-fseeko.c) = ab62b8c7faa9a601191475c040c7466e1bd52fe1
+SHA1 (patch-aq) = 02186df2372b3317cc2df6a8230cc198ac3eb4e9
diff --git a/sysutils/coreutils/patches/patch-aq b/sysutils/coreutils/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/sysutils/coreutils/patches/patch-aq
@@ -0,0 +1,23 @@
+diff -r fb0a1c120d18 lib/fseterr.c
+--- lib/fseterr.c	Sat May 05 16:00:33 2012 +0000
++++ lib/fseterr.c	Sat May 05 16:05:52 2012 +0000
+@@ -71,6 +71,6 @@
+     }
+   errno = saved_errno;
+ #else
+- #error "Please port gnulib fseterr.c to your platform! Look at the definitions of ferror and clearerr on your system, then report this to bug-gnulib."
++  return 1;
+ #endif
+ }
+diff -r fb0a1c120d18 src/dd.c
+--- src/dd.c	Sat May 05 16:00:33 2012 +0000
++++ src/dd.c	Sat May 05 16:05:52 2012 +0000
+@@ -1375,7 +1375,7 @@
+    The offending behavior has been confirmed with an Exabyte SCSI tape
+    drive accessed via /dev/nst0 on both Linux 2.2.17 and 2.4.16 kernels.  */
+ 
+-#ifdef __linux__
++#if 0
+ 
+ # include <sys/mtio.h>
+ 
