# HG changeset patch
# Parent 6d53010f381dc5d84487dc09296de852fc365079

diff --git a/textproc/groff/PLIST b/textproc/groff/PLIST
--- a/textproc/groff/PLIST
+++ b/textproc/groff/PLIST
@@ -15,9 +15,6 @@
 bin/grops
 bin/grotty
 bin/hpftodit
-bin/indxbib
-bin/lkbib
-bin/lookbib
 bin/mmroff
 bin/neqn
 bin/nroff
@@ -28,7 +25,6 @@
 bin/post-grohtml
 bin/pre-grohtml
 bin/preconv
-bin/refer
 bin/roff2dvi
 bin/roff2html
 bin/roff2pdf
@@ -40,11 +36,11 @@
 bin/tfmtodit
 bin/troff
 info/groff.info
-lib/groff/groffer/func.pl
-lib/groff/groffer/man.pl
-lib/groff/groffer/perl_test.pl
-lib/groff/groffer/split_env.sh
-lib/groff/groffer/version.sh
+@comment lib/groff/groffer/func.pl
+@comment lib/groff/groffer/man.pl
+@comment lib/groff/groffer/perl_test.pl
+@comment lib/groff/groffer/split_env.sh
+@comment lib/groff/groffer/version.sh
 man/man1/addftinfo.1
 man/man1/afmtodit.1
 man/man1/chem.1
