# HG changeset patch
# Parent 783a8ec4eaf9519b0957ca626c77eef2e19535f4

diff --git a/textproc/hunspell/distinfo b/textproc/hunspell/distinfo
--- a/textproc/hunspell/distinfo
+++ b/textproc/hunspell/distinfo
@@ -6,3 +6,4 @@
 SHA1 (patch-aa) = b03a51d88bd57786f4a0ac01a3b760b4d8ba385a
 SHA1 (patch-ab) = 01ff90fc2ebf1049d791a0669ddcc3f5495729c0
 SHA1 (patch-ac) = 27cad6078b55bce7144f462ddebac9887392c93b
+SHA1 (patch-aq) = ac8d1f4468d017c841d33ab7cb073944921f03bf
diff --git a/textproc/hunspell/patches/patch-aq b/textproc/hunspell/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/textproc/hunspell/patches/patch-aq
@@ -0,0 +1,24 @@
+diff -r 006c3f933ef3 src/tools/munch.c
+--- src/tools/munch.c	Mon May 07 23:49:28 2012 +0000
++++ src/tools/munch.c	Mon May 07 23:50:25 2012 +0000
+@@ -8,7 +8,7 @@
+ #include <sys/types.h>
+ #include <sys/stat.h>
+ #include <fcntl.h>
+-#ifdef __linux__
++#if defined(__linux__) && defined(__GLIBC__)
+ #include <error.h>
+ #include <errno.h>
+ #include <sys/mman.h>
+diff -r 006c3f933ef3 src/tools/unmunch.c
+--- src/tools/unmunch.c	Mon May 07 23:49:28 2012 +0000
++++ src/tools/unmunch.c	Mon May 07 23:50:25 2012 +0000
+@@ -10,7 +10,7 @@
+ #include <sys/types.h>
+ #include <sys/stat.h>
+ #include <fcntl.h>
+-#ifdef __linux__
++#if defined(__linux__) && defined(__GLIBC__)
+ #include <error.h>
+ #include <errno.h>
+ #include <sys/mman.h>
