# HG changeset patch
# Parent e45956a5469b16d92cbb36010753e07d2cf098c0

diff --git a/textproc/xmlrpc-c/Makefile b/textproc/xmlrpc-c/Makefile
--- a/textproc/xmlrpc-c/Makefile
+++ b/textproc/xmlrpc-c/Makefile
@@ -40,5 +40,7 @@
 
 .include "../../www/curl/buildlink3.mk"
 .include "../../www/libwww/buildlink3.mk"
+.include "../../converters/libiconv/buildlink3.mk"
+.include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../mk/pthread.buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
diff --git a/textproc/xmlrpc-c/distinfo b/textproc/xmlrpc-c/distinfo
--- a/textproc/xmlrpc-c/distinfo
+++ b/textproc/xmlrpc-c/distinfo
@@ -13,3 +13,4 @@
 SHA1 (patch-lib_curl__transport_curlmulti.c) = 1cea4c43b1ed3dcf73827afc14d7bb9267371121
 SHA1 (patch-lib_curl__transport_curltransaction.c) = 1dcf46c0dd1dd8e7fdfbb6c81e00b326f6d0b1f9
 SHA1 (patch-lib_curl__transport_xmlrpc__curl__transport.c) = ea132ebb46ff0ce9893d64e9c9205d5a155220ef
+SHA1 (patch-aq) = 434c86fea4f13a8d0ba92e82b11bf3476a3017da
diff --git a/textproc/xmlrpc-c/patches/patch-aq b/textproc/xmlrpc-c/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/textproc/xmlrpc-c/patches/patch-aq
@@ -0,0 +1,48 @@
+diff -r 89442310b346 examples/interrupted_client.c
+--- examples/interrupted_client.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/interrupted_client.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -2,6 +2,8 @@
+    both by timeout and by control-C.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <unistd.h>
+ #include <stdio.h>
+diff -r 89442310b346 examples/interrupted_server.c
+--- examples/interrupted_server.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/interrupted_server.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -7,6 +7,8 @@
+    simpler code, but it is not interruptible with SIGTERM.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <signal.h>
+diff -r 89442310b346 examples/xmlrpc_inetd_server.c
+--- examples/xmlrpc_inetd_server.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/xmlrpc_inetd_server.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -21,6 +21,8 @@
+    respond to the client, then exit.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <signal.h>
+diff -r 89442310b346 examples/xmlrpc_loop_server.c
+--- examples/xmlrpc_loop_server.c	Mon Jul 02 16:31:43 2012 +0000
++++ examples/xmlrpc_loop_server.c	Mon Jul 02 16:34:32 2012 +0000
+@@ -6,6 +6,8 @@
+    less control over how the requests are served.
+ */
+ 
++#define _GNU_SOURCE
++
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <signal.h>
