# HG changeset patch
# Parent 232b5a0a643da8cd8fb5c7285b11953992487315
diff --git a/textproc/libxml2/Makefile b/textproc/libxml2/Makefile
--- a/textproc/libxml2/Makefile
+++ b/textproc/libxml2/Makefile
@@ -25,6 +25,9 @@
 CONFIGURE_ARGS+=	--with-html-subdir=libxml2
 CONFIGURE_ARGS+=	--without-python
 
+# [Gregor] Threading support is broken on musl
+CONFIGURE_ARGS+=	--without-threads
+
 .include "options.mk"
 
 # allow thread-awareness, but make sure the library is not
