# HG changeset patch
# Parent c420d1a9a9ead7e193822bc109f68bb256edd3ab

diff --git a/www/firefox/Makefile b/www/firefox/Makefile
--- a/www/firefox/Makefile
+++ b/www/firefox/Makefile
@@ -20,6 +20,7 @@
 
 LDFLAGS.DragonFly=	-lplc4 -lnspr4
 LDFLAGS.FreeBSD=	-lplc4 -lnspr4
+LDFLAGS.Linux=	-lplc4 -lnspr4
 
 .include "../../devel/xulrunner/mozilla-common.mk"
 .include "options.mk"
@@ -32,6 +33,9 @@
 MOZILLA_NAME=	Browser
 .endif
 
+# [Gregor] To make some patches work
+CFLAGS+=		-D__musl__
+
 pre-configure:
 	cd ${WRKSRC} && autoconf
 # XXX not built here, just to make check-portability.awk shut up.
