# HG changeset patch
# Parent 72e6acea2cbd2b0d018886fb29c2993d4b01070b

diff --git a/x11/libFS/Makefile b/x11/libFS/Makefile
--- a/x11/libFS/Makefile
+++ b/x11/libFS/Makefile
@@ -27,6 +27,8 @@
 
 .include "../../meta-pkgs/modular-xorg/hacks.mk"
 
+CONFIGURE_ARGS+=	CFLAGS="-O2 -g -DNOFILES_MAX=256"
+
 .include "../../x11/fontsproto/buildlink3.mk"
 .include "../../x11/xproto/buildlink3.mk"
 .include "../../x11/xtrans/buildlink3.mk"
diff --git a/x11/libFS/distinfo b/x11/libFS/distinfo
--- a/x11/libFS/distinfo
+++ b/x11/libFS/distinfo
@@ -3,3 +3,4 @@
 SHA1 (libFS-1.0.4.tar.bz2) = f114f32f60eb84d3e1d0e6f0dfe7679b07e91295
 RMD160 (libFS-1.0.4.tar.bz2) = 6e74cc9b4d8ece4abcabb26e3c5cec7a0e591918
 Size (libFS-1.0.4.tar.bz2) = 291155 bytes
+SHA1 (patch-aq) = 75f64d23fe0ffdc5dca48fcba982539d34caf42b
diff --git a/x11/libFS/patches/patch-aq b/x11/libFS/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/libFS/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 5f55f742a811 src/FSConnServ.c
+--- src/FSConnServ.c	Mon May 07 19:40:35 2012 +0000
++++ src/FSConnServ.c	Mon May 07 19:44:53 2012 +0000
+@@ -49,6 +49,8 @@
+ 
+ */
+ 
++#define _GNU_SOURCE
++
+ #ifdef HAVE_CONFIG_H
+ #include <config.h>
+ #endif
