# HG changeset patch
# Parent 8f3ed01a2e2600e112e34d1262e6abdadafdc838

diff --git a/x11/pixman/Makefile b/x11/pixman/Makefile
--- a/x11/pixman/Makefile
+++ b/x11/pixman/Makefile
@@ -42,5 +42,8 @@
 post-extract:
 	${CP} ${FILESDIR}/pixman.sym ${WRKDIR}
 	${CP} ${FILESDIR}/solaris-hwcap.mapfile ${WRKSRC}/pixman
-
+	
+pre-configure:
+	sed -i 's/support_for__thread=yes/support_for__thread=no/' ${WRKSRC}/configure	
+	
 .include "../../mk/bsd.pkg.mk"
