# HG changeset patch
# Parent 82a123b06e24f86ca2e48c6d49e0283b26ef6aae
diff --git a/x11/modular-xorg-server/distinfo b/x11/modular-xorg-server/distinfo
--- a/x11/modular-xorg-server/distinfo
+++ b/x11/modular-xorg-server/distinfo
@@ -33,3 +33,4 @@
 SHA1 (patch-se) = 22b20c734b7fcf59d75227aeb1fa703465582568
 SHA1 (patch-sf) = 22d4f575c066691696ec96f1fcd459251b662012
 SHA1 (patch-sg) = 81598e687357a6e8b477055d14ece01372ae4617
+SHA1 (patch-zz) = e7c951b98794b4d4a32279198cf82f6378aa7d79
diff --git a/x11/modular-xorg-server/patches/patch-zz b/x11/modular-xorg-server/patches/patch-zz
new file mode 100644
--- /dev/null
+++ b/x11/modular-xorg-server/patches/patch-zz
@@ -0,0 +1,58 @@
+diff -r 64d7c5193c40 hw/xfree86/common/xf86AutoConfig.c
+--- hw/xfree86/common/xf86AutoConfig.c	Thu Jul 19 15:50:42 2012 +0000
++++ hw/xfree86/common/xf86AutoConfig.c	Thu Jul 19 15:51:21 2012 +0000
+@@ -368,7 +368,7 @@
+                 goto end;
+             }
+             /* Read the file */
+-#ifdef __GLIBC__
++#if defined(__GLIBC__) || 1
+             while ((read = getline(&line, &len, fp)) != -1) {
+ #else
+             while ((line = fgetln(fp, &len)) != (char *)NULL) {
+diff -r 64d7c5193c40 hw/xfree86/os-support/linux/lnx_video.c
+--- hw/xfree86/os-support/linux/lnx_video.c	Thu Jul 19 15:50:42 2012 +0000
++++ hw/xfree86/os-support/linux/lnx_video.c	Thu Jul 19 15:51:21 2012 +0000
+@@ -53,6 +53,7 @@
+ 
+ static Bool ExtendedEnabled = FALSE;
+ 
++#if 0
+ #ifdef __ia64__
+ 
+ #include "compiler.h"
+@@ -73,6 +74,13 @@
+ 
+ #endif
+ 
++#else
++#include <sys/syscall.h>
++#define ioperm(x, y, z) syscall(__NR_ioperm, x, y, z)
++#define iopl(x) syscall(__NR_iopl, x)
++
++#endif
++
+ #ifdef __alpha__
+ 
+ extern void sethae(unsigned long hae);
+diff -r 64d7c5193c40 hw/xfree86/os-support/xf86_OSlib.h
+--- hw/xfree86/os-support/xf86_OSlib.h	Thu Jul 19 15:50:42 2012 +0000
++++ hw/xfree86/os-support/xf86_OSlib.h	Thu Jul 19 15:51:21 2012 +0000
+@@ -215,6 +215,7 @@
+ # include <sys/types.h>
+ # include <assert.h>
+ 
++#if 0
+ # ifdef __linux__
+ #  include <termio.h>
+ # else /* __GLIBC__ */
+@@ -223,6 +224,9 @@
+ # ifdef __sparc__
+ #  include <sys/param.h>
+ # endif
++#else
++#include <termios.h>
++#endif
+ 
+ # include <errno.h>
+ 
