# HG changeset patch
# Parent 5cdb4c7bdeee84c07f6c950484bba51c8466960b

diff --git a/x11/xload/distinfo b/x11/xload/distinfo
--- a/x11/xload/distinfo
+++ b/x11/xload/distinfo
@@ -3,3 +3,4 @@
 SHA1 (xload-1.1.1.tar.bz2) = 50f6ee958e6fb609af92cf9403e7bd68720575cb
 RMD160 (xload-1.1.1.tar.bz2) = 8c971e743cbd3e7cfe8ac2747d87ae2d4c0981d5
 Size (xload-1.1.1.tar.bz2) = 130887 bytes
+SHA1 (patch-aq) = 0b45fe95dc54313063cf5ab8be83b3c657274c47
diff --git a/x11/xload/patches/patch-aq b/x11/xload/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/xload/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r f6e1349acbf9 get_rload.c
+--- get_rload.c	Thu Jul 19 23:47:59 2012 +0000
++++ get_rload.c	Thu Jul 19 23:49:09 2012 +0000
+@@ -10,7 +10,7 @@
+ /* Not all OS supports get_rload
+    steal the STUB idea from get_load
+  */
+-#if defined(QNX4) || defined(__CYGWIN__)
++#if defined(QNX4) || defined(__CYGWIN__) || 1
+ #define RLOADSTUB
+ #endif
+ 
