# HG changeset patch
# Parent 94db6594f0c7578308bbb006efc6862f84e6837c

diff --git a/x11/libFS/Makefile b/x11/libFS/Makefile
--- a/x11/libFS/Makefile
+++ b/x11/libFS/Makefile
@@ -25,6 +25,8 @@
 CONFIGURE_ARGS+=	--disable-malloc0returnsnull
 .endif
 
+CONFIGURE_ARGS+=	CFLAGS="-O2 -g -DNOFILES_MAX=256"
+
 .include "../../x11/fontsproto/buildlink3.mk"
 .include "../../x11/xproto/buildlink3.mk"
 .include "../../x11/xtrans/buildlink3.mk"
diff --git a/x11/libFS/distinfo b/x11/libFS/distinfo
--- a/x11/libFS/distinfo
+++ b/x11/libFS/distinfo
@@ -3,3 +3,4 @@
 SHA1 (libFS-1.0.3.tar.bz2) = 2b33fa17369605303ae70a7de1aa681879249c92
 RMD160 (libFS-1.0.3.tar.bz2) = a4c1811de0cba5fd40a083031c44a951c240dd3b
 Size (libFS-1.0.3.tar.bz2) = 284331 bytes
+SHA1 (patch-aq) = 75f64d23fe0ffdc5dca48fcba982539d34caf42b
diff --git a/x11/libFS/patches/patch-aq b/x11/libFS/patches/patch-aq
new file mode 100644
--- /dev/null
+++ b/x11/libFS/patches/patch-aq
@@ -0,0 +1,12 @@
+diff -r 5f55f742a811 src/FSConnServ.c
+--- src/FSConnServ.c	Mon May 07 19:40:35 2012 +0000
++++ src/FSConnServ.c	Mon May 07 19:44:53 2012 +0000
+@@ -49,6 +49,8 @@
+ 
+ */
+ 
++#define _GNU_SOURCE
++
+ #ifdef HAVE_CONFIG_H
+ #include <config.h>
+ #endif
